Search.setIndex({"docnames": ["gtdblib", "gtdblib.cli", "gtdblib.db", "gtdblib.exception", "gtdblib.file", "gtdblib.file.itol", "gtdblib.taxon", "gtdblib.taxonomy", "gtdblib.tree", "gtdblib.util", "gtdblib.util.bio", "gtdblib.util.shell", "index", "modules"], "filenames": ["gtdblib.rst", "gtdblib.cli.rst", "gtdblib.db.rst", "gtdblib.exception.rst", "gtdblib.file.rst", "gtdblib.file.itol.rst", "gtdblib.taxon.rst", "gtdblib.taxonomy.rst", "gtdblib.tree.rst", "gtdblib.util.rst", "gtdblib.util.bio.rst", "gtdblib.util.shell.rst", "index.rst", "modules.rst"], "titles": ["gtdblib package", "gtdblib.cli package", "gtdblib.db package", "gtdblib.exception package", "gtdblib.file package", "gtdblib.file.itol package", "gtdblib.taxon package", "gtdblib.taxonomy package", "gtdblib.tree package", "gtdblib.util package", "gtdblib.util.bio package", "gtdblib.util.shell package", "Welcome to GTDB Library\u2019s documentation!", "gtdblib"], "terms": {"cli": [0, 12, 13], "tree": [0, 5, 7, 10, 11, 12, 13], "bootstrap_merg": [0, 1, 12, 13], "convert_to_canon": [0, 1], "polytomi": [0, 1, 12, 13], "db": [0, 12, 13], "common": [0, 7, 12, 13], "metadata": [0, 2], "registri": [0, 2], "type_annotation_map": [0, 2], "lpsnhtml": [0, 2], "assigned_bi": [0, 2], "assigned_by_doi": [0, 2], "basonym": [0, 2], "categori": [0, 2], "creat": [0, 2, 10, 11], "effective_publ": [0, 2], "effective_publication_doi": [0, 2], "emend": [0, 2], "etymologi": [0, 2], "gender": [0, 2], "html": [0, 2], "id": [0, 2, 7, 10], "ijsem_list": [0, 2], "ijsem_list_doi": [0, 2], "n_child_correct": [0, 2], "n_child_synonym": [0, 2], "n_child_tot": [0, 2], "name": [0, 2, 7, 10, 11], "nomenclatural_statu": [0, 2], "original_publ": [0, 2], "original_publication_doi": [0, 2], "parent_taxon": [0, 2], "proposed_a": [0, 2], "record_numb": [0, 2], "risk_group": [0, 2], "ssu": [0, 2], "ssu_ebi": [0, 2], "ssu_fasta": [0, 2], "ssu_ggdc": [0, 2], "ssu_ncbi": [0, 2], "strain_info": [0, 2], "taxonomic_statu": [0, 2], "to_process": [0, 2], "tyg": [0, 2], "type_class": [0, 2], "type_genu": [0, 2], "type_ord": [0, 2], "type_speci": [0, 2], "type_strain": [0, 2], "type_subgenu": [0, 2], "updat": [0, 2], "url": [0, 2], "valid_publ": [0, 2], "valid_publication_doi": [0, 2], "lpsnhtmlchildtaxa": [0, 2], "child_page_id": [0, 2], "parent_page_id": [0, 2], "lpsnhtmlnote": [0, 2], "doi": [0, 2], "note": [0, 2], "page_id": [0, 2], "lpsnhtmlsynonym": [0, 2], "kind": [0, 2], "synonym_id": [0, 2], "genericengin": [0, 2], "except": [0, 9, 12, 13], "gtdblibexcept": [0, 3], "gtdblibexit": [0, 3], "file": [0, 7, 10, 11, 12, 13], "itol": [0, 4], "collaps": [0, 1, 4, 8], "dataset_color_strip": [0, 4], "dataset_text": [0, 4], "label": [0, 4, 7, 10], "popup": [0, 4], "tree_color": [0, 4], "path": [0, 1, 4, 5, 8, 11], "taxon": [0, 7, 8, 10, 12, 13], "rank": [0, 7, 12, 13], "taxonrank": [0, 6], "class": [0, 2, 4, 5, 6, 7], "domain": [0, 6, 7], "famili": [0, 6, 7], "genu": [0, 6, 7], "order": [0, 6, 7, 10], "phylum": [0, 6, 7], "speci": [0, 6, 7], "valu": [0, 1, 5, 6, 8, 10, 11], "taxonomi": [0, 12, 13], "class_index": [0, 7], "domain_index": [0, 7], "family_index": [0, 7], "genus_index": [0, 7], "order_index": [0, 7], "phylum_index": [0, 7], "rank_index": [0, 7], "rank_label": [0, 7], "rank_prefix": [0, 7], "species_index": [0, 7], "c": [0, 7, 10], "d": [0, 7, 10], "f": [0, 7], "g": [0, 1, 6, 7, 9, 10, 11], "o": [0, 7], "p": [0, 7], "": [0, 7, 9], "fill_trailing_rank": [0, 7], "read_taxonomi": [0, 7], "read_taxonomy_from_tre": [0, 7], "valid": [0, 12, 13], "duplicate_nam": [0, 7], "taxonomic_consist": [0, 7], "validate_species_nam": [0, 7], "validate_taxonomi": [0, 7], "bootstrap_merge_repl": [0, 8], "convert_access": [0, 12, 13], "convert_tree_accessions_to_canon": [0, 8], "get_node_depth": [0, 12, 13], "get_tree_node_depth": [0, 8], "get_node_to_desc_taxa": [0, 12, 13], "get_tree_node_to_desc_taxa": [0, 8], "get_unique_mrca_for_internal_nod": [0, 12, 13], "get_tree_unique_mrca_for_internal_nod": [0, 8], "collapse_polytomi": [0, 8], "collapse_polytomy_tre": [0, 8], "util": [0, 12, 13], "bio": [0, 9], "access": [0, 1, 7, 8, 9], "newick": [0, 7, 9], "seq_io": [0, 9], "shell": [0, 9], "color": [0, 5, 12, 13], "execut": [0, 9], "filemgmt": [0, 9], "gtdbshutil": [0, 9], "rgb_to_hex": [0, 9, 11], "type": [0, 1, 10, 12, 13], "is_float": [0, 9], "pathlib": 1, "typer": 1, "model": 1, "argumentinfo": 1, "object": [1, 2, 4, 6, 7, 8, 9], "output": [1, 10, 11], "replic": 1, "list": [1, 5, 7, 11], "cpu": [1, 8], "int": [1, 2, 5, 8, 9, 10, 11], "optioninfo": 1, "calcul": [1, 8, 10, 11], "non": [1, 8], "parametr": [1, 8], "bootstrap": [1, 8], "us": [1, 2, 7, 8, 10], "multipl": 1, "convert": [1, 7, 8, 9, 11], "canon": [1, 7, 8, 10], "form": [1, 7, 8, 10], "e": [1, 5, 6, 7, 8, 10], "gca_123456789": 1, "1": [1, 5, 7, 8, 10, 11], "g123456789": 1, "support": [1, 8, 10], "float": [1, 5, 8, 9, 10], "node": [1, 5, 8], "low": [1, 8], "kwarg": 2, "ani": [2, 7, 9, 11], "base": [2, 3, 4, 5, 6, 7, 10, 11], "declarativebas": 2, "classvar": 2, "refer": 2, "_schema": 2, "collect": [2, 8], "new": 2, "tabl": 2, "orm_declarative_metadata": 2, "_registrytyp": 2, "sqlalchemi": 2, "orm": 2, "decl_api": 2, "_orm": 2, "where": [2, 7], "mapper": 2, "associ": 2, "datetim": 2, "timestamp": 2, "timezon": 2, "true": [2, 5, 7, 10, 11], "map": [2, 7], "str": [2, 3, 5, 6, 7, 8, 9, 10, 11], "bool": [2, 5, 7, 9, 10, 11], "db_name": 2, "properti": 2, "messag": [3, 7], "all": [3, 4, 5, 7, 8, 11], "gtdb": 3, "lib": 3, "thrown": 3, "thi": [3, 7, 8, 10, 11], "project": 3, "rais": [3, 11], "when": [3, 11], "ha": [3, 10], "been": 3, "ask": 3, "quietli": 3, "exit": [3, 11], "submodul": [4, 12, 13], "itolcollapsefil": [4, 5], "data": [4, 5], "insert": [4, 5, 10], "write": [4, 5, 8, 10], "itoldatasetcolorstripfil": [4, 5], "itoldatasettextfil": [4, 5], "itollabelfil": [4, 5], "itolpopupfil": [4, 5], "itoltreecolorsfil": [4, 5], "insert_branch": [4, 5], "insert_clad": [4, 5], "insert_label": [4, 5], "insert_label_background": [4, 5], "insert_rang": [4, 5], "The": [4, 8, 10, 11], "inherit": 4, "from": [4, 7, 8, 10, 11], "node_id": 5, "color_branch": 5, "fals": [5, 7, 10, 11], "strip_width": 5, "25": 5, "margin": 5, "0": [5, 7, 10], "border_width": 5, "border_color": 5, "0000ff": 5, "complete_bord": 5, "show_intern": 5, "show_strip_label": 5, "strip_label_posit": 5, "liter": 5, "top": 5, "center": 5, "bottom": 5, "strip_label_size_factor": 5, "strip_label_rot": 5, "strip_label_shift": 5, "strip_label_color": 5, "000000": 5, "strip_label_outlin": 5, "5": [5, 7], "show_label": 5, "size_factor": 5, "label_rot": 5, "label_shift": 5, "labels_below": 5, "vertical_shift": 5, "straight_label": 5, "align_to_tre": 5, "external_label_shift": 5, "label_size_factor": 5, "posit": 5, "style": [5, 7], "normal": 5, "bold": 5, "ital": 5, "rotat": 5, "defin": [5, 10], "text": 5, "extern": [5, 11], "number": [5, 7, 8, 10, 11], "between": 5, "intern": [5, 7, 8], "specifi": [5, 7], "along": [5, 10], "branch": 5, "exampl": [5, 7, 10], "i": [5, 7, 8, 10, 11], "exactli": 5, "start": [5, 7], "middl": 5, "end": 5, "titl": 5, "branch_styl": 5, "scale_factor": 5, "singl": 5, "clade": 5, "font_styl": 5, "option": [5, 10], "none": [5, 7, 10, 11], "font": 5, "leaf": 5, "background": 5, "rang": 5, "enum": 6, "A": [6, 7, 8, 10], "taxonom": [6, 7], "etc": 6, "c__": [6, 7], "d__": [6, 7], "f__": [6, 7], "g__": [6, 7], "o__": [6, 7], "p__": [6, 7], "s__": [6, 7], "2": 7, "4": 7, "3": 7, "6": 7, "taxa": [7, 8], "fill": 7, "miss": 7, "trail": 7, "string": [7, 9, 10, 11], "function": [7, 10, 11], "assum": [7, 10], "provid": [7, 10], "7": 7, "ar": [7, 8, 10, 11], "d__bacteria": 7, "d__firmicut": 7, "taxonomy_fil": 7, "use_canonical_gid": 7, "dict": [7, 8, 10], "read": [7, 9, 10], "greengen": 7, "method": [7, 10], "gener": [7, 10], "can": [7, 9, 11], "doe": [7, 11], "strictli": 7, "requir": 7, "seven": 7, "present": 7, "expect": 7, "format": [7, 10, 11], "typic": 7, "have": [7, 8, 10], "indic": 7, "assign": 7, "flag": 7, "should": [7, 10], "warn": 7, "obtain": 7, "each": [7, 8, 10], "extant": 7, "extran": 7, "classifi": 7, "theseven": 7, "filenam": 7, "dendropi": [7, 8], "issu": 7, "log": 7, "extent": 7, "check_speci": 7, "identifi": [7, 8], "duplic": 7, "unique_id": 7, "index": [7, 10, 12], "uniqu": [7, 8, 10], "lineag": 7, "report_error": 7, "determin": [7, 10], "consist": 7, "classif": 7, "boolean": [7, 10], "error": [7, 11], "written": 7, "screen": 7, "parent": 7, "inconsist": 7, "species_nam": 7, "require_ful": 7, "require_prefix": 7, "tupl": [7, 8, 10, 11], "full": 7, "binomi": 7, "includ": 7, "specif": [7, 10, 11], "epithet": 7, "escherichia": 7, "coli": 7, "two": 7, "separ": 7, "space": [7, 10], "must": 7, "prefix": 7, "otherwis": 7, "reason": 7, "fail": 7, "check_prefix": 7, "check_rank": 7, "check_hierarchi": 7, "check_group_nam": 7, "check_duplicate_nam": 7, "check_capit": 7, "check": [7, 9, 10, 11], "strict": 7, "hierarchi": 7, "implement": [7, 11], "workflow": 7, "set": [7, 8, 10], "It": [7, 10], "configur": 7, "input": [7, 8, 10], "presenc": 7, "group": [7, 8], "invalid": 7, "charact": [7, 10], "taxon_id": 7, "child_taxon_id": 7, "more": [7, 10], "input_tre": 8, "output_tre": 8, "replicate_tre": 8, "monophylet": 8, "transpos": 8, "them": 8, "paramet": [8, 9, 10, 11], "merg": 8, "process": [8, 11], "input_path": 8, "output_path": 8, "depth": 8, "return": [8, 10, 11], "dictionari": [8, 10], "frozenset": 8, "get": [8, 10], "descend": 8, "mrca": 8, "alwai": 8, "yield": 8, "same": [8, 10], "path_in": 8, "path_out": 8, "less": 8, "than": 8, "work": 8, "canonical_gid": [9, 10], "is_same_accn_vers": [9, 10], "bioerror": [9, 10], "create_label": [9, 10], "parse_label": [9, 10], "inputfileerror": [9, 10], "extract_seq": [9, 10], "is_nucleotid": [9, 10], "is_protein": [9, 10], "read_fasta": [9, 10], "read_fasta_seq": [9, 10], "read_fastq": [9, 10], "read_fastq_seq": [9, 10], "read_seq": [9, 10], "seq_length": [9, 10], "write_fasta": [9, 10], "hex_to_rgb": [9, 11], "check_depend": [9, 11], "check_on_path": [9, 11], "is_execut": [9, 11], "run": [9, 11], "run_bash": [9, 11], "which": [9, 10, 11], "matching_bracket": [9, 11], "select_delimit": [9, 11], "sha256": [9, 11], "sha256_rb": [9, 11], "check_dir_exist": [9, 11], "check_file_exist": [9, 11], "get_num_lin": [9, 11], "make_sure_path_exist": [9, 11], "symlink": [9, 11], "r": [9, 11], "b": [9, 11], "gid": 10, "ncbi": 10, "genom": 10, "g005435135": 10, "gcf_005435135": 10, "1_asm543513v1_genom": 10, "rs_gcf_005435135": 10, "gb_gca_005435135": 10, "acces": 10, "conver": 10, "accn1": 10, "accn2": 10, "version": 10, "suffix": 10, "proceed": 10, "period": 10, "first": 10, "second": 10, "els": [10, 11], "msg": 10, "auxiliary_info": 10, "auxiliari": 10, "inform": 10, "pars": 10, "mai": 10, "contain": 10, "fasta_fil": 10, "seqs_to_extract": 10, "extract": 10, "sequenc": 10, "fasta": 10, "seq_id": 10, "seq": 10, "seq_fil": 10, "req_perc": 10, "95": 10, "max_seqs_to_read": 10, "10": [10, 11], "nucleotid": 10, "perform": 10, "look": 10, "t": [10, 11], "n": 10, "confirm": 10, "compris": 10, "major": 10, "primarili": 10, "q": 10, "percentag": 10, "befor": 10, "declar": 10, "being": 10, "maximum": 10, "protein": 10, "20": 10, "amino": 10, "acid": 10, "x": 10, "keep_annot": 10, "annot": 10, "intend": 10, "avoid": 10, "larg": 10, "memori": 10, "gzip": 10, "seqio": 10, "print": 10, "iter": 10, "follow": 10, "itself": 10, "fastq_fil": 10, "fastq": 10, "either": 10, "slightli": 10, "effici": 10, "directli": 10, "call": 10, "known": 10, "param": [10, 11], "length": 10, "output_fil": 10, "wrap": 10, "80": 10, "If": [10, 11], "extens": 10, "gz": 10, "compress": 10, "produc": 10, "per": 10, "line": [10, 11], "modifi": 11, "hex": 11, "rgb": 11, "red": 11, "green": 11, "blue": 11, "program": 11, "exit_on_fail": 11, "system": 11, "code": 11, "fpath": 11, "cmd": 11, "command": 11, "stdout": 11, "via": 11, "bash": 11, "python": 11, "linux": 11, "http": 11, "stackoverflow": 11, "com": 11, "question": 11, "377017": 11, "test": 11, "exist": 11, "isn": 11, "split": 11, "match": 11, "bracket": 11, "substr": 11, "so": 11, "we": 11, "nest": 11, "input_fil": 11, "select": 11, "delimit": 11, "csv": 11, "tsv": 11, "hash": 11, "binari": 11, "copi": 11, "archiv": 11, "directori": 11, "input_dir": 11, "biolib": 11, "09": 11, "2020": 11, "file_path": 11, "wa": 11, "target": 11, "link_nam": 11, "overwrit": 11, "symbol": 11, "link": 11, "point": 11, "fileexistserror": 11, "unless": 11, "try": 11, "isadirectoryerror": 11, "gtdblib": 12, "packag": [12, 13], "subpackag": [12, 13], "modul": [12, 13], "config": [12, 13], "search": 12, "page": 12, "content": 13}, "objects": {"": [[0, 0, 0, "-", "gtdblib"]], "gtdblib": [[1, 0, 0, "-", "cli"], [0, 0, 0, "-", "config"], [2, 0, 0, "-", "db"], [3, 0, 0, "-", "exception"], [4, 0, 0, "-", "file"], [6, 0, 0, "-", "taxon"], [7, 0, 0, "-", "taxonomy"], [8, 0, 0, "-", "tree"], [9, 0, 0, "-", "util"]], "gtdblib.cli": [[1, 0, 0, "-", "tree"]], "gtdblib.cli.tree": [[1, 1, 1, "", "bootstrap_merge"], [1, 1, 1, "", "convert_to_canonical"], [1, 1, 1, "", "polytomy"]], "gtdblib.db": [[2, 2, 1, "", "GenericEngine"], [2, 0, 0, "-", "common"]], "gtdblib.db.GenericEngine": [[2, 3, 1, "", "url"]], "gtdblib.db.common": [[2, 2, 1, "", "Common"], [2, 2, 1, "", "LpsnHtml"], [2, 2, 1, "", "LpsnHtmlChildTaxa"], [2, 2, 1, "", "LpsnHtmlNotes"], [2, 2, 1, "", "LpsnHtmlSynonyms"]], "gtdblib.db.common.Common": [[2, 4, 1, "", "metadata"], [2, 4, 1, "", "registry"], [2, 4, 1, "", "type_annotation_map"]], "gtdblib.db.common.LpsnHtml": [[2, 4, 1, "", "assigned_by"], [2, 4, 1, "", "assigned_by_doi"], [2, 4, 1, "", "basonym"], [2, 4, 1, "", "category"], [2, 4, 1, "", "created"], [2, 4, 1, "", "effective_publication"], [2, 4, 1, "", "effective_publication_doi"], [2, 4, 1, "", "emendations"], [2, 4, 1, "", "etymology"], [2, 4, 1, "", "gender"], [2, 4, 1, "", "html"], [2, 4, 1, "", "id"], [2, 4, 1, "", "ijsem_list"], [2, 4, 1, "", "ijsem_list_doi"], [2, 4, 1, "", "n_child_correct"], [2, 4, 1, "", "n_child_synonym"], [2, 4, 1, "", "n_child_total"], [2, 4, 1, "", "name"], [2, 4, 1, "", "nomenclatural_status"], [2, 4, 1, "", "original_publication"], [2, 4, 1, "", "original_publication_doi"], [2, 4, 1, "", "parent_taxon"], [2, 4, 1, "", "proposed_as"], [2, 4, 1, "", "record_number"], [2, 4, 1, "", "risk_group"], [2, 4, 1, "", "ssu"], [2, 4, 1, "", "ssu_ebi"], [2, 4, 1, "", "ssu_fasta"], [2, 4, 1, "", "ssu_ggdc"], [2, 4, 1, "", "ssu_ncbi"], [2, 4, 1, "", "strain_info"], [2, 4, 1, "", "taxonomic_status"], [2, 4, 1, "", "to_process"], [2, 4, 1, "", "tygs"], [2, 4, 1, "", "type_class"], [2, 4, 1, "", "type_genus"], [2, 4, 1, "", "type_order"], [2, 4, 1, "", "type_species"], [2, 4, 1, "", "type_strain"], [2, 4, 1, "", "type_subgenus"], [2, 4, 1, "", "updated"], [2, 4, 1, "", "url"], [2, 4, 1, "", "valid_publication"], [2, 4, 1, "", "valid_publication_doi"]], "gtdblib.db.common.LpsnHtmlChildTaxa": [[2, 4, 1, "", "child_page_id"], [2, 4, 1, "", "id"], [2, 4, 1, "", "nomenclatural_status"], [2, 4, 1, "", "parent_page_id"], [2, 4, 1, "", "taxonomic_status"]], "gtdblib.db.common.LpsnHtmlNotes": [[2, 4, 1, "", "doi"], [2, 4, 1, "", "id"], [2, 4, 1, "", "note"], [2, 4, 1, "", "page_id"]], "gtdblib.db.common.LpsnHtmlSynonyms": [[2, 4, 1, "", "id"], [2, 4, 1, "", "kind"], [2, 4, 1, "", "page_id"], [2, 4, 1, "", "synonym_id"]], "gtdblib.exception": [[3, 5, 1, "", "GtdbLibException"], [3, 5, 1, "", "GtdbLibExit"]], "gtdblib.file": [[4, 2, 1, "", "File"], [5, 0, 0, "-", "itol"]], "gtdblib.file.File": [[4, 4, 1, "", "path"]], "gtdblib.file.itol": [[5, 0, 0, "-", "collapse"], [5, 0, 0, "-", "dataset_color_strip"], [5, 0, 0, "-", "dataset_text"], [5, 0, 0, "-", "label"], [5, 0, 0, "-", "popup"], [5, 0, 0, "-", "tree_colors"]], "gtdblib.file.itol.collapse": [[5, 2, 1, "", "iTolCollapseFile"]], "gtdblib.file.itol.collapse.iTolCollapseFile": [[5, 4, 1, "", "data"], [5, 6, 1, "", "insert"], [5, 6, 1, "", "write"]], "gtdblib.file.itol.dataset_color_strip": [[5, 2, 1, "", "iTolDatasetColorStripFile"]], "gtdblib.file.itol.dataset_color_strip.iTolDatasetColorStripFile": [[5, 4, 1, "", "data"], [5, 6, 1, "", "insert"], [5, 6, 1, "", "write"]], "gtdblib.file.itol.dataset_text": [[5, 2, 1, "", "iTolDatasetTextFile"]], "gtdblib.file.itol.dataset_text.iTolDatasetTextFile": [[5, 4, 1, "", "data"], [5, 6, 1, "", "insert"], [5, 6, 1, "", "write"]], "gtdblib.file.itol.label": [[5, 2, 1, "", "iTolLabelFile"]], "gtdblib.file.itol.label.iTolLabelFile": [[5, 4, 1, "", "data"], [5, 6, 1, "", "insert"], [5, 6, 1, "", "write"]], "gtdblib.file.itol.popup": [[5, 2, 1, "", "iTolPopupFile"]], "gtdblib.file.itol.popup.iTolPopupFile": [[5, 4, 1, "", "data"], [5, 6, 1, "", "insert"], [5, 6, 1, "", "write"]], "gtdblib.file.itol.tree_colors": [[5, 2, 1, "", "iTolTreeColorsFile"]], "gtdblib.file.itol.tree_colors.iTolTreeColorsFile": [[5, 4, 1, "", "data"], [5, 6, 1, "", "insert_branch"], [5, 6, 1, "", "insert_clade"], [5, 6, 1, "", "insert_label"], [5, 6, 1, "", "insert_label_background"], [5, 6, 1, "", "insert_range"], [5, 6, 1, "", "write"]], "gtdblib.taxon": [[6, 0, 0, "-", "rank"], [6, 0, 0, "-", "taxon"]], "gtdblib.taxon.rank": [[6, 2, 1, "", "TaxonRank"]], "gtdblib.taxon.rank.TaxonRank": [[6, 4, 1, "", "CLASS"], [6, 4, 1, "", "DOMAIN"], [6, 4, 1, "", "FAMILY"], [6, 4, 1, "", "GENUS"], [6, 4, 1, "", "ORDER"], [6, 4, 1, "", "PHYLUM"], [6, 4, 1, "", "SPECIES"]], "gtdblib.taxon.taxon": [[6, 2, 1, "", "Taxon"]], "gtdblib.taxon.taxon.Taxon": [[6, 4, 1, "", "rank"], [6, 4, 1, "", "value"]], "gtdblib.taxonomy": [[7, 0, 0, "-", "taxonomy"], [7, 0, 0, "-", "validation"]], "gtdblib.taxonomy.taxonomy": [[7, 2, 1, "", "Taxonomy"], [7, 1, 1, "", "fill_trailing_ranks"], [7, 1, 1, "", "read_taxonomy"], [7, 1, 1, "", "read_taxonomy_from_tree"]], "gtdblib.taxonomy.taxonomy.Taxonomy": [[7, 4, 1, "", "CLASS_INDEX"], [7, 4, 1, "", "DOMAIN_INDEX"], [7, 4, 1, "", "FAMILY_INDEX"], [7, 4, 1, "", "GENUS_INDEX"], [7, 4, 1, "", "ORDER_INDEX"], [7, 4, 1, "", "PHYLUM_INDEX"], [7, 4, 1, "", "RANK_INDEX"], [7, 4, 1, "", "RANK_LABELS"], [7, 4, 1, "", "RANK_PREFIXES"], [7, 4, 1, "", "SPECIES_INDEX"], [7, 4, 1, "", "c"], [7, 4, 1, "", "d"], [7, 4, 1, "", "f"], [7, 4, 1, "", "g"], [7, 4, 1, "", "o"], [7, 4, 1, "", "p"], [7, 4, 1, "", "s"]], "gtdblib.taxonomy.validation": [[7, 1, 1, "", "duplicate_names"], [7, 1, 1, "", "taxonomic_consistency"], [7, 1, 1, "", "validate_species_name"], [7, 1, 1, "", "validate_taxonomy"]], "gtdblib.tree": [[8, 0, 0, "-", "bootstrap_merge"], [8, 0, 0, "-", "convert_accession"], [8, 0, 0, "-", "get_node_depth"], [8, 0, 0, "-", "get_node_to_desc_taxa"], [8, 0, 0, "-", "get_unique_mrca_for_internal_nodes"], [8, 0, 0, "-", "polytomy"]], "gtdblib.tree.bootstrap_merge": [[8, 1, 1, "", "bootstrap_merge_replicates"]], "gtdblib.tree.convert_accession": [[8, 1, 1, "", "convert_tree_accessions_to_canonical"]], "gtdblib.tree.get_node_depth": [[8, 1, 1, "", "get_tree_node_depth"]], "gtdblib.tree.get_node_to_desc_taxa": [[8, 1, 1, "", "get_tree_node_to_desc_taxa"]], "gtdblib.tree.get_unique_mrca_for_internal_nodes": [[8, 1, 1, "", "get_tree_unique_mrca_for_internal_nodes"]], "gtdblib.tree.polytomy": [[8, 1, 1, "", "collapse_polytomy"], [8, 1, 1, "", "collapse_polytomy_tree"]], "gtdblib.util": [[10, 0, 0, "-", "bio"], [9, 0, 0, "-", "color"], [11, 0, 0, "-", "shell"], [9, 0, 0, "-", "types"]], "gtdblib.util.bio": [[10, 0, 0, "-", "accession"], [10, 0, 0, "-", "exceptions"], [10, 0, 0, "-", "newick"], [10, 0, 0, "-", "seq_io"]], "gtdblib.util.bio.accession": [[10, 1, 1, "", "canonical_gid"], [10, 1, 1, "", "is_same_accn_version"]], "gtdblib.util.bio.exceptions": [[10, 5, 1, "", "BioError"]], "gtdblib.util.bio.newick": [[10, 1, 1, "", "create_label"], [10, 1, 1, "", "parse_label"]], "gtdblib.util.bio.seq_io": [[10, 5, 1, "", "InputFileError"], [10, 1, 1, "", "extract_seqs"], [10, 1, 1, "", "is_nucleotide"], [10, 1, 1, "", "is_protein"], [10, 1, 1, "", "read"], [10, 1, 1, "", "read_fasta"], [10, 1, 1, "", "read_fasta_seq"], [10, 1, 1, "", "read_fastq"], [10, 1, 1, "", "read_fastq_seq"], [10, 1, 1, "", "read_seq"], [10, 1, 1, "", "seq_lengths"], [10, 1, 1, "", "write_fasta"]], "gtdblib.util.color": [[9, 1, 1, "", "rgb_to_hex"]], "gtdblib.util.shell": [[11, 0, 0, "-", "color"], [11, 0, 0, "-", "execute"], [11, 0, 0, "-", "filemgmt"], [11, 0, 0, "-", "gtdbshutil"]], "gtdblib.util.shell.color": [[11, 1, 1, "", "hex_to_rgb"], [11, 1, 1, "", "rgb_to_hex"]], "gtdblib.util.shell.execute": [[11, 1, 1, "", "check_dependencies"], [11, 1, 1, "", "check_on_path"], [11, 1, 1, "", "is_executable"], [11, 1, 1, "", "run"], [11, 1, 1, "", "run_bash"], [11, 1, 1, "", "which"]], "gtdblib.util.shell.filemgmt": [[11, 1, 1, "", "matching_brackets"], [11, 1, 1, "", "select_delimiter"], [11, 1, 1, "", "sha256"], [11, 1, 1, "", "sha256_rb"]], "gtdblib.util.shell.gtdbshutil": [[11, 1, 1, "", "check_dir_exists"], [11, 1, 1, "", "check_file_exists"], [11, 1, 1, "", "get_num_lines"], [11, 1, 1, "", "make_sure_path_exists"], [11, 1, 1, "", "symlink"]], "gtdblib.util.types": [[9, 1, 1, "", "is_float"]]}, "objtypes": {"0": "py:module", "1": "py:function", "2": "py:class", "3": "py:property", "4": "py:attribute", "5": "py:exception", "6": "py:method"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "function", "Python function"], "2": ["py", "class", "Python class"], "3": ["py", "property", "Python property"], "4": ["py", "attribute", "Python attribute"], "5": ["py", "exception", "Python exception"], "6": ["py", "method", "Python method"]}, "titleterms": {"gtdblib": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 13], "packag": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11], "subpackag": [0, 4, 9], "submodul": [0, 1, 2, 5, 6, 7, 8, 9, 10, 11], "config": 0, "modul": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11], "content": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12], "cli": 1, "tree": [1, 8], "db": 2, "common": 2, "except": [3, 10], "file": [4, 5], "itol": 5, "collaps": 5, "dataset_color_strip": 5, "dataset_text": 5, "label": 5, "popup": 5, "tree_color": 5, "taxon": 6, "rank": 6, "taxonomi": 7, "valid": 7, "paramet": 7, "return": 7, "bootstrap_merg": 8, "convert_access": 8, "get_node_depth": 8, "get_node_to_desc_taxa": 8, "get_unique_mrca_for_internal_nod": 8, "polytomi": 8, "util": [9, 10, 11], "color": [9, 11], "type": 9, "bio": 10, "access": 10, "newick": 10, "seq_io": 10, "shell": 11, "execut": 11, "filemgmt": 11, "gtdbshutil": 11, "welcom": 12, "gtdb": 12, "librari": 12, "": 12, "document": 12, "indic": 12, "tabl": 12}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx": 57}, "alltitles": {"gtdblib package": [[0, "gtdblib-package"]], "Subpackages": [[0, "subpackages"], [4, "subpackages"], [9, "subpackages"]], "Submodules": [[0, "submodules"], [1, "submodules"], [2, "submodules"], [5, "submodules"], [6, "submodules"], [7, "submodules"], [8, "submodules"], [9, "submodules"], [10, "submodules"], [11, "submodules"]], "gtdblib.config module": [[0, "module-gtdblib.config"]], "Module contents": [[0, "module-gtdblib"], [1, "module-gtdblib.cli"], [2, "module-gtdblib.db"], [3, "module-gtdblib.exception"], [4, "module-gtdblib.file"], [5, "module-gtdblib.file.itol"], [6, "module-gtdblib.taxon"], [7, "module-gtdblib.taxonomy"], [8, "module-gtdblib.tree"], [9, "module-gtdblib.util"], [10, "module-gtdblib.util.bio"], [11, "module-gtdblib.util.shell"]], "gtdblib.cli package": [[1, "gtdblib-cli-package"]], "gtdblib.cli.tree module": [[1, "module-gtdblib.cli.tree"]], "gtdblib.db package": [[2, "gtdblib-db-package"]], "gtdblib.db.common module": [[2, "module-gtdblib.db.common"]], "gtdblib.exception package": [[3, "gtdblib-exception-package"]], "gtdblib.file package": [[4, "gtdblib-file-package"]], "gtdblib.file.itol package": [[5, "gtdblib-file-itol-package"]], "gtdblib.file.itol.collapse module": [[5, "module-gtdblib.file.itol.collapse"]], "gtdblib.file.itol.dataset_color_strip module": [[5, "module-gtdblib.file.itol.dataset_color_strip"]], "gtdblib.file.itol.dataset_text module": [[5, "module-gtdblib.file.itol.dataset_text"]], "gtdblib.file.itol.label module": [[5, "module-gtdblib.file.itol.label"]], "gtdblib.file.itol.popup module": [[5, "module-gtdblib.file.itol.popup"]], "gtdblib.file.itol.tree_colors module": [[5, "module-gtdblib.file.itol.tree_colors"]], "gtdblib.taxon package": [[6, "gtdblib-taxon-package"]], "gtdblib.taxon.rank module": [[6, "module-gtdblib.taxon.rank"]], "gtdblib.taxon.taxon module": [[6, "module-gtdblib.taxon.taxon"]], "gtdblib.taxonomy package": [[7, "gtdblib-taxonomy-package"]], "gtdblib.taxonomy.taxonomy module": [[7, "module-gtdblib.taxonomy.taxonomy"]], "gtdblib.taxonomy.validation module": [[7, "module-gtdblib.taxonomy.validation"]], "Parameters": [[7, "parameters"], [7, "id1"], [7, "id3"], [7, "id5"]], "Returns": [[7, "returns"], [7, "id2"], [7, "id4"], [7, "id6"]], "gtdblib.tree package": [[8, "gtdblib-tree-package"]], "gtdblib.tree.bootstrap_merge module": [[8, "module-gtdblib.tree.bootstrap_merge"]], "gtdblib.tree.convert_accession module": [[8, "module-gtdblib.tree.convert_accession"]], "gtdblib.tree.get_node_depth module": [[8, "module-gtdblib.tree.get_node_depth"]], "gtdblib.tree.get_node_to_desc_taxa module": [[8, "module-gtdblib.tree.get_node_to_desc_taxa"]], "gtdblib.tree.get_unique_mrca_for_internal_nodes module": [[8, "module-gtdblib.tree.get_unique_mrca_for_internal_nodes"]], "gtdblib.tree.polytomy module": [[8, "module-gtdblib.tree.polytomy"]], "gtdblib.util package": [[9, "gtdblib-util-package"]], "gtdblib.util.color module": [[9, "module-gtdblib.util.color"]], "gtdblib.util.types module": [[9, "module-gtdblib.util.types"]], "gtdblib.util.bio package": [[10, "gtdblib-util-bio-package"]], "gtdblib.util.bio.accession module": [[10, "module-gtdblib.util.bio.accession"]], "gtdblib.util.bio.exceptions module": [[10, "module-gtdblib.util.bio.exceptions"]], "gtdblib.util.bio.newick module": [[10, "module-gtdblib.util.bio.newick"]], "gtdblib.util.bio.seq_io module": [[10, "module-gtdblib.util.bio.seq_io"]], "gtdblib.util.shell package": [[11, "gtdblib-util-shell-package"]], "gtdblib.util.shell.color module": [[11, "module-gtdblib.util.shell.color"]], "gtdblib.util.shell.execute module": [[11, "module-gtdblib.util.shell.execute"]], "gtdblib.util.shell.filemgmt module": [[11, "module-gtdblib.util.shell.filemgmt"]], "gtdblib.util.shell.gtdbshutil module": [[11, "module-gtdblib.util.shell.gtdbshutil"]], "Welcome to GTDB Library\u2019s documentation!": [[12, "welcome-to-gtdb-library-s-documentation"]], "Contents:": [[12, null]], "Indices and tables": [[12, "indices-and-tables"]], "gtdblib": [[13, "gtdblib"]]}, "indexentries": {"gtdblib": [[0, "module-gtdblib"]], "gtdblib.config": [[0, "module-gtdblib.config"]], "module": [[0, "module-gtdblib"], [0, "module-gtdblib.config"], [1, "module-gtdblib.cli"], [1, "module-gtdblib.cli.tree"], [2, "module-gtdblib.db"], [2, "module-gtdblib.db.common"], [3, "module-gtdblib.exception"], [4, "module-gtdblib.file"], [5, "module-gtdblib.file.itol"], [5, "module-gtdblib.file.itol.collapse"], [5, "module-gtdblib.file.itol.dataset_color_strip"], [5, "module-gtdblib.file.itol.dataset_text"], [5, "module-gtdblib.file.itol.label"], [5, "module-gtdblib.file.itol.popup"], [5, "module-gtdblib.file.itol.tree_colors"], [6, "module-gtdblib.taxon"], [6, "module-gtdblib.taxon.rank"], [6, "module-gtdblib.taxon.taxon"], [7, "module-gtdblib.taxonomy"], [7, "module-gtdblib.taxonomy.taxonomy"], [7, "module-gtdblib.taxonomy.validation"], [8, "module-gtdblib.tree"], [8, "module-gtdblib.tree.bootstrap_merge"], [8, "module-gtdblib.tree.convert_accession"], [8, "module-gtdblib.tree.get_node_depth"], [8, "module-gtdblib.tree.get_node_to_desc_taxa"], [8, "module-gtdblib.tree.get_unique_mrca_for_internal_nodes"], [8, "module-gtdblib.tree.polytomy"], [9, "module-gtdblib.util"], [9, "module-gtdblib.util.color"], [9, "module-gtdblib.util.types"], [10, "module-gtdblib.util.bio"], [10, "module-gtdblib.util.bio.accession"], [10, "module-gtdblib.util.bio.exceptions"], [10, "module-gtdblib.util.bio.newick"], [10, "module-gtdblib.util.bio.seq_io"], [11, "module-gtdblib.util.shell"], [11, "module-gtdblib.util.shell.color"], [11, "module-gtdblib.util.shell.execute"], [11, "module-gtdblib.util.shell.filemgmt"], [11, "module-gtdblib.util.shell.gtdbshutil"]], "bootstrap_merge() (in module gtdblib.cli.tree)": [[1, "gtdblib.cli.tree.bootstrap_merge"]], "convert_to_canonical() (in module gtdblib.cli.tree)": [[1, "gtdblib.cli.tree.convert_to_canonical"]], "gtdblib.cli": [[1, "module-gtdblib.cli"]], "gtdblib.cli.tree": [[1, "module-gtdblib.cli.tree"]], "polytomy() (in module gtdblib.cli.tree)": [[1, "gtdblib.cli.tree.polytomy"]], "common (class in gtdblib.db.common)": [[2, "gtdblib.db.common.Common"]], "genericengine (class in gtdblib.db)": [[2, "gtdblib.db.GenericEngine"]], "lpsnhtml (class in gtdblib.db.common)": [[2, "gtdblib.db.common.LpsnHtml"]], "lpsnhtmlchildtaxa (class in gtdblib.db.common)": [[2, "gtdblib.db.common.LpsnHtmlChildTaxa"]], "lpsnhtmlnotes (class in gtdblib.db.common)": [[2, "gtdblib.db.common.LpsnHtmlNotes"]], "lpsnhtmlsynonyms (class in gtdblib.db.common)": [[2, "gtdblib.db.common.LpsnHtmlSynonyms"]], "assigned_by (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.assigned_by"]], "assigned_by_doi (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.assigned_by_doi"]], "basonym (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.basonym"]], "category (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.category"]], "child_page_id (gtdblib.db.common.lpsnhtmlchildtaxa attribute)": [[2, "gtdblib.db.common.LpsnHtmlChildTaxa.child_page_id"]], "created (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.created"]], "doi (gtdblib.db.common.lpsnhtmlnotes attribute)": [[2, "gtdblib.db.common.LpsnHtmlNotes.doi"]], "effective_publication (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.effective_publication"]], "effective_publication_doi (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.effective_publication_doi"]], "emendations (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.emendations"]], "etymology (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.etymology"]], "gender (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.gender"]], "gtdblib.db": [[2, "module-gtdblib.db"]], "gtdblib.db.common": [[2, "module-gtdblib.db.common"]], "html (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.html"]], "id (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.id"]], "id (gtdblib.db.common.lpsnhtmlchildtaxa attribute)": [[2, "gtdblib.db.common.LpsnHtmlChildTaxa.id"]], "id (gtdblib.db.common.lpsnhtmlnotes attribute)": [[2, "gtdblib.db.common.LpsnHtmlNotes.id"]], "id (gtdblib.db.common.lpsnhtmlsynonyms attribute)": [[2, "gtdblib.db.common.LpsnHtmlSynonyms.id"]], "ijsem_list (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.ijsem_list"]], "ijsem_list_doi (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.ijsem_list_doi"]], "kind (gtdblib.db.common.lpsnhtmlsynonyms attribute)": [[2, "gtdblib.db.common.LpsnHtmlSynonyms.kind"]], "metadata (gtdblib.db.common.common attribute)": [[2, "gtdblib.db.common.Common.metadata"]], "n_child_correct (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.n_child_correct"]], "n_child_synonym (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.n_child_synonym"]], "n_child_total (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.n_child_total"]], "name (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.name"]], "nomenclatural_status (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.nomenclatural_status"]], "nomenclatural_status (gtdblib.db.common.lpsnhtmlchildtaxa attribute)": [[2, "gtdblib.db.common.LpsnHtmlChildTaxa.nomenclatural_status"]], "note (gtdblib.db.common.lpsnhtmlnotes attribute)": [[2, "gtdblib.db.common.LpsnHtmlNotes.note"]], "original_publication (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.original_publication"]], "original_publication_doi (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.original_publication_doi"]], "page_id (gtdblib.db.common.lpsnhtmlnotes attribute)": [[2, "gtdblib.db.common.LpsnHtmlNotes.page_id"]], "page_id (gtdblib.db.common.lpsnhtmlsynonyms attribute)": [[2, "gtdblib.db.common.LpsnHtmlSynonyms.page_id"]], "parent_page_id (gtdblib.db.common.lpsnhtmlchildtaxa attribute)": [[2, "gtdblib.db.common.LpsnHtmlChildTaxa.parent_page_id"]], "parent_taxon (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.parent_taxon"]], "proposed_as (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.proposed_as"]], "record_number (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.record_number"]], "registry (gtdblib.db.common.common attribute)": [[2, "gtdblib.db.common.Common.registry"]], "risk_group (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.risk_group"]], "ssu (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.ssu"]], "ssu_ebi (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.ssu_ebi"]], "ssu_fasta (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.ssu_fasta"]], "ssu_ggdc (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.ssu_ggdc"]], "ssu_ncbi (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.ssu_ncbi"]], "strain_info (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.strain_info"]], "synonym_id (gtdblib.db.common.lpsnhtmlsynonyms attribute)": [[2, "gtdblib.db.common.LpsnHtmlSynonyms.synonym_id"]], "taxonomic_status (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.taxonomic_status"]], "taxonomic_status (gtdblib.db.common.lpsnhtmlchildtaxa attribute)": [[2, "gtdblib.db.common.LpsnHtmlChildTaxa.taxonomic_status"]], "to_process (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.to_process"]], "tygs (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.tygs"]], "type_annotation_map (gtdblib.db.common.common attribute)": [[2, "gtdblib.db.common.Common.type_annotation_map"]], "type_class (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.type_class"]], "type_genus (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.type_genus"]], "type_order (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.type_order"]], "type_species (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.type_species"]], "type_strain (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.type_strain"]], "type_subgenus (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.type_subgenus"]], "updated (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.updated"]], "url (gtdblib.db.genericengine property)": [[2, "gtdblib.db.GenericEngine.url"]], "url (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.url"]], "valid_publication (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.valid_publication"]], "valid_publication_doi (gtdblib.db.common.lpsnhtml attribute)": [[2, "gtdblib.db.common.LpsnHtml.valid_publication_doi"]], "gtdblibexception": [[3, "gtdblib.exception.GtdbLibException"]], "gtdblibexit": [[3, "gtdblib.exception.GtdbLibExit"]], "gtdblib.exception": [[3, "module-gtdblib.exception"]], "file (class in gtdblib.file)": [[4, "gtdblib.file.File"]], "gtdblib.file": [[4, "module-gtdblib.file"]], "path (gtdblib.file.file attribute)": [[4, "gtdblib.file.File.path"]], "data (gtdblib.file.itol.collapse.itolcollapsefile attribute)": [[5, "gtdblib.file.itol.collapse.iTolCollapseFile.data"]], "data (gtdblib.file.itol.dataset_color_strip.itoldatasetcolorstripfile attribute)": [[5, "gtdblib.file.itol.dataset_color_strip.iTolDatasetColorStripFile.data"]], "data (gtdblib.file.itol.dataset_text.itoldatasettextfile attribute)": [[5, "gtdblib.file.itol.dataset_text.iTolDatasetTextFile.data"]], "data (gtdblib.file.itol.label.itollabelfile attribute)": [[5, "gtdblib.file.itol.label.iTolLabelFile.data"]], "data (gtdblib.file.itol.popup.itolpopupfile attribute)": [[5, "gtdblib.file.itol.popup.iTolPopupFile.data"]], "data (gtdblib.file.itol.tree_colors.itoltreecolorsfile attribute)": [[5, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.data"]], "gtdblib.file.itol": [[5, "module-gtdblib.file.itol"]], "gtdblib.file.itol.collapse": [[5, "module-gtdblib.file.itol.collapse"]], "gtdblib.file.itol.dataset_color_strip": [[5, "module-gtdblib.file.itol.dataset_color_strip"]], "gtdblib.file.itol.dataset_text": [[5, "module-gtdblib.file.itol.dataset_text"]], "gtdblib.file.itol.label": [[5, "module-gtdblib.file.itol.label"]], "gtdblib.file.itol.popup": [[5, "module-gtdblib.file.itol.popup"]], "gtdblib.file.itol.tree_colors": [[5, "module-gtdblib.file.itol.tree_colors"]], "itolcollapsefile (class in gtdblib.file.itol.collapse)": [[5, "gtdblib.file.itol.collapse.iTolCollapseFile"]], "itoldatasetcolorstripfile (class in gtdblib.file.itol.dataset_color_strip)": [[5, "gtdblib.file.itol.dataset_color_strip.iTolDatasetColorStripFile"]], "itoldatasettextfile (class in gtdblib.file.itol.dataset_text)": [[5, "gtdblib.file.itol.dataset_text.iTolDatasetTextFile"]], "itollabelfile (class in gtdblib.file.itol.label)": [[5, "gtdblib.file.itol.label.iTolLabelFile"]], "itolpopupfile (class in gtdblib.file.itol.popup)": [[5, "gtdblib.file.itol.popup.iTolPopupFile"]], "itoltreecolorsfile (class in gtdblib.file.itol.tree_colors)": [[5, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile"]], "insert() (gtdblib.file.itol.collapse.itolcollapsefile method)": [[5, "gtdblib.file.itol.collapse.iTolCollapseFile.insert"]], "insert() (gtdblib.file.itol.dataset_color_strip.itoldatasetcolorstripfile method)": [[5, "gtdblib.file.itol.dataset_color_strip.iTolDatasetColorStripFile.insert"]], "insert() (gtdblib.file.itol.dataset_text.itoldatasettextfile method)": [[5, "gtdblib.file.itol.dataset_text.iTolDatasetTextFile.insert"]], "insert() (gtdblib.file.itol.label.itollabelfile method)": [[5, "gtdblib.file.itol.label.iTolLabelFile.insert"]], "insert() (gtdblib.file.itol.popup.itolpopupfile method)": [[5, "gtdblib.file.itol.popup.iTolPopupFile.insert"]], "insert_branch() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[5, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.insert_branch"]], "insert_clade() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[5, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.insert_clade"]], "insert_label() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[5, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.insert_label"]], "insert_label_background() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[5, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.insert_label_background"]], "insert_range() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[5, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.insert_range"]], "write() (gtdblib.file.itol.collapse.itolcollapsefile method)": [[5, "gtdblib.file.itol.collapse.iTolCollapseFile.write"]], "write() (gtdblib.file.itol.dataset_color_strip.itoldatasetcolorstripfile method)": [[5, "gtdblib.file.itol.dataset_color_strip.iTolDatasetColorStripFile.write"]], "write() (gtdblib.file.itol.dataset_text.itoldatasettextfile method)": [[5, "gtdblib.file.itol.dataset_text.iTolDatasetTextFile.write"]], "write() (gtdblib.file.itol.label.itollabelfile method)": [[5, "gtdblib.file.itol.label.iTolLabelFile.write"]], "write() (gtdblib.file.itol.popup.itolpopupfile method)": [[5, "gtdblib.file.itol.popup.iTolPopupFile.write"]], "write() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[5, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.write"]], "class (gtdblib.taxon.rank.taxonrank attribute)": [[6, "gtdblib.taxon.rank.TaxonRank.CLASS"]], "domain (gtdblib.taxon.rank.taxonrank attribute)": [[6, "gtdblib.taxon.rank.TaxonRank.DOMAIN"]], "family (gtdblib.taxon.rank.taxonrank attribute)": [[6, "gtdblib.taxon.rank.TaxonRank.FAMILY"]], "genus (gtdblib.taxon.rank.taxonrank attribute)": [[6, "gtdblib.taxon.rank.TaxonRank.GENUS"]], "order (gtdblib.taxon.rank.taxonrank attribute)": [[6, "gtdblib.taxon.rank.TaxonRank.ORDER"]], "phylum (gtdblib.taxon.rank.taxonrank attribute)": [[6, "gtdblib.taxon.rank.TaxonRank.PHYLUM"]], "species (gtdblib.taxon.rank.taxonrank attribute)": [[6, "gtdblib.taxon.rank.TaxonRank.SPECIES"]], "taxon (class in gtdblib.taxon.taxon)": [[6, "gtdblib.taxon.taxon.Taxon"]], "taxonrank (class in gtdblib.taxon.rank)": [[6, "gtdblib.taxon.rank.TaxonRank"]], "gtdblib.taxon": [[6, "module-gtdblib.taxon"]], "gtdblib.taxon.rank": [[6, "module-gtdblib.taxon.rank"]], "gtdblib.taxon.taxon": [[6, "module-gtdblib.taxon.taxon"]], "rank (gtdblib.taxon.taxon.taxon attribute)": [[6, "gtdblib.taxon.taxon.Taxon.rank"]], "value (gtdblib.taxon.taxon.taxon attribute)": [[6, "gtdblib.taxon.taxon.Taxon.value"]], "class_index (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.CLASS_INDEX"]], "domain_index (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.DOMAIN_INDEX"]], "family_index (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.FAMILY_INDEX"]], "genus_index (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.GENUS_INDEX"]], "order_index (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.ORDER_INDEX"]], "phylum_index (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.PHYLUM_INDEX"]], "rank_index (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.RANK_INDEX"]], "rank_labels (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.RANK_LABELS"]], "rank_prefixes (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.RANK_PREFIXES"]], "species_index (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.SPECIES_INDEX"]], "taxonomy (class in gtdblib.taxonomy.taxonomy)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy"]], "c (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.c"]], "d (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.d"]], "duplicate_names() (in module gtdblib.taxonomy.validation)": [[7, "gtdblib.taxonomy.validation.duplicate_names"]], "f (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.f"]], "fill_trailing_ranks() (in module gtdblib.taxonomy.taxonomy)": [[7, "gtdblib.taxonomy.taxonomy.fill_trailing_ranks"]], "g (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.g"]], "gtdblib.taxonomy": [[7, "module-gtdblib.taxonomy"]], "gtdblib.taxonomy.taxonomy": [[7, "module-gtdblib.taxonomy.taxonomy"]], "gtdblib.taxonomy.validation": [[7, "module-gtdblib.taxonomy.validation"]], "o (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.o"]], "p (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.p"]], "read_taxonomy() (in module gtdblib.taxonomy.taxonomy)": [[7, "gtdblib.taxonomy.taxonomy.read_taxonomy"]], "read_taxonomy_from_tree() (in module gtdblib.taxonomy.taxonomy)": [[7, "gtdblib.taxonomy.taxonomy.read_taxonomy_from_tree"]], "s (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[7, "gtdblib.taxonomy.taxonomy.Taxonomy.s"]], "taxonomic_consistency() (in module gtdblib.taxonomy.validation)": [[7, "gtdblib.taxonomy.validation.taxonomic_consistency"]], "validate_species_name() (in module gtdblib.taxonomy.validation)": [[7, "gtdblib.taxonomy.validation.validate_species_name"]], "validate_taxonomy() (in module gtdblib.taxonomy.validation)": [[7, "gtdblib.taxonomy.validation.validate_taxonomy"]], "bootstrap_merge_replicates() (in module gtdblib.tree.bootstrap_merge)": [[8, "gtdblib.tree.bootstrap_merge.bootstrap_merge_replicates"]], "collapse_polytomy() (in module gtdblib.tree.polytomy)": [[8, "gtdblib.tree.polytomy.collapse_polytomy"]], "collapse_polytomy_tree() (in module gtdblib.tree.polytomy)": [[8, "gtdblib.tree.polytomy.collapse_polytomy_tree"]], "convert_tree_accessions_to_canonical() (in module gtdblib.tree.convert_accession)": [[8, "gtdblib.tree.convert_accession.convert_tree_accessions_to_canonical"]], "get_tree_node_depth() (in module gtdblib.tree.get_node_depth)": [[8, "gtdblib.tree.get_node_depth.get_tree_node_depth"]], "get_tree_node_to_desc_taxa() (in module gtdblib.tree.get_node_to_desc_taxa)": [[8, "gtdblib.tree.get_node_to_desc_taxa.get_tree_node_to_desc_taxa"]], "get_tree_unique_mrca_for_internal_nodes() (in module gtdblib.tree.get_unique_mrca_for_internal_nodes)": [[8, "gtdblib.tree.get_unique_mrca_for_internal_nodes.get_tree_unique_mrca_for_internal_nodes"]], "gtdblib.tree": [[8, "module-gtdblib.tree"]], "gtdblib.tree.bootstrap_merge": [[8, "module-gtdblib.tree.bootstrap_merge"]], "gtdblib.tree.convert_accession": [[8, "module-gtdblib.tree.convert_accession"]], "gtdblib.tree.get_node_depth": [[8, "module-gtdblib.tree.get_node_depth"]], "gtdblib.tree.get_node_to_desc_taxa": [[8, "module-gtdblib.tree.get_node_to_desc_taxa"]], "gtdblib.tree.get_unique_mrca_for_internal_nodes": [[8, "module-gtdblib.tree.get_unique_mrca_for_internal_nodes"]], "gtdblib.tree.polytomy": [[8, "module-gtdblib.tree.polytomy"]], "gtdblib.util": [[9, "module-gtdblib.util"]], "gtdblib.util.color": [[9, "module-gtdblib.util.color"]], "gtdblib.util.types": [[9, "module-gtdblib.util.types"]], "is_float() (in module gtdblib.util.types)": [[9, "gtdblib.util.types.is_float"]], "rgb_to_hex() (in module gtdblib.util.color)": [[9, "gtdblib.util.color.rgb_to_hex"]], "bioerror": [[10, "gtdblib.util.bio.exceptions.BioError"]], "inputfileerror": [[10, "gtdblib.util.bio.seq_io.InputFileError"]], "canonical_gid() (in module gtdblib.util.bio.accession)": [[10, "gtdblib.util.bio.accession.canonical_gid"]], "create_label() (in module gtdblib.util.bio.newick)": [[10, "gtdblib.util.bio.newick.create_label"]], "extract_seqs() (in module gtdblib.util.bio.seq_io)": [[10, "gtdblib.util.bio.seq_io.extract_seqs"]], "gtdblib.util.bio": [[10, "module-gtdblib.util.bio"]], "gtdblib.util.bio.accession": [[10, "module-gtdblib.util.bio.accession"]], "gtdblib.util.bio.exceptions": [[10, "module-gtdblib.util.bio.exceptions"]], "gtdblib.util.bio.newick": [[10, "module-gtdblib.util.bio.newick"]], "gtdblib.util.bio.seq_io": [[10, "module-gtdblib.util.bio.seq_io"]], "is_nucleotide() (in module gtdblib.util.bio.seq_io)": [[10, "gtdblib.util.bio.seq_io.is_nucleotide"]], "is_protein() (in module gtdblib.util.bio.seq_io)": [[10, "gtdblib.util.bio.seq_io.is_protein"]], "is_same_accn_version() (in module gtdblib.util.bio.accession)": [[10, "gtdblib.util.bio.accession.is_same_accn_version"]], "parse_label() (in module gtdblib.util.bio.newick)": [[10, "gtdblib.util.bio.newick.parse_label"]], "read() (in module gtdblib.util.bio.seq_io)": [[10, "gtdblib.util.bio.seq_io.read"]], "read_fasta() (in module gtdblib.util.bio.seq_io)": [[10, "gtdblib.util.bio.seq_io.read_fasta"]], "read_fasta_seq() (in module gtdblib.util.bio.seq_io)": [[10, "gtdblib.util.bio.seq_io.read_fasta_seq"]], "read_fastq() (in module gtdblib.util.bio.seq_io)": [[10, "gtdblib.util.bio.seq_io.read_fastq"]], "read_fastq_seq() (in module gtdblib.util.bio.seq_io)": [[10, "gtdblib.util.bio.seq_io.read_fastq_seq"]], "read_seq() (in module gtdblib.util.bio.seq_io)": [[10, "gtdblib.util.bio.seq_io.read_seq"]], "seq_lengths() (in module gtdblib.util.bio.seq_io)": [[10, "gtdblib.util.bio.seq_io.seq_lengths"]], "write_fasta() (in module gtdblib.util.bio.seq_io)": [[10, "gtdblib.util.bio.seq_io.write_fasta"]], "check_dependencies() (in module gtdblib.util.shell.execute)": [[11, "gtdblib.util.shell.execute.check_dependencies"]], "check_dir_exists() (in module gtdblib.util.shell.gtdbshutil)": [[11, "gtdblib.util.shell.gtdbshutil.check_dir_exists"]], "check_file_exists() (in module gtdblib.util.shell.gtdbshutil)": [[11, "gtdblib.util.shell.gtdbshutil.check_file_exists"]], "check_on_path() (in module gtdblib.util.shell.execute)": [[11, "gtdblib.util.shell.execute.check_on_path"]], "get_num_lines() (in module gtdblib.util.shell.gtdbshutil)": [[11, "gtdblib.util.shell.gtdbshutil.get_num_lines"]], "gtdblib.util.shell": [[11, "module-gtdblib.util.shell"]], "gtdblib.util.shell.color": [[11, "module-gtdblib.util.shell.color"]], "gtdblib.util.shell.execute": [[11, "module-gtdblib.util.shell.execute"]], "gtdblib.util.shell.filemgmt": [[11, "module-gtdblib.util.shell.filemgmt"]], "gtdblib.util.shell.gtdbshutil": [[11, "module-gtdblib.util.shell.gtdbshutil"]], "hex_to_rgb() (in module gtdblib.util.shell.color)": [[11, "gtdblib.util.shell.color.hex_to_rgb"]], "is_executable() (in module gtdblib.util.shell.execute)": [[11, "gtdblib.util.shell.execute.is_executable"]], "make_sure_path_exists() (in module gtdblib.util.shell.gtdbshutil)": [[11, "gtdblib.util.shell.gtdbshutil.make_sure_path_exists"]], "matching_brackets() (in module gtdblib.util.shell.filemgmt)": [[11, "gtdblib.util.shell.filemgmt.matching_brackets"]], "rgb_to_hex() (in module gtdblib.util.shell.color)": [[11, "gtdblib.util.shell.color.rgb_to_hex"]], "run() (in module gtdblib.util.shell.execute)": [[11, "gtdblib.util.shell.execute.run"]], "run_bash() (in module gtdblib.util.shell.execute)": [[11, "gtdblib.util.shell.execute.run_bash"]], "select_delimiter() (in module gtdblib.util.shell.filemgmt)": [[11, "gtdblib.util.shell.filemgmt.select_delimiter"]], "sha256() (in module gtdblib.util.shell.filemgmt)": [[11, "gtdblib.util.shell.filemgmt.sha256"]], "sha256_rb() (in module gtdblib.util.shell.filemgmt)": [[11, "gtdblib.util.shell.filemgmt.sha256_rb"]], "symlink() (in module gtdblib.util.shell.gtdbshutil)": [[11, "gtdblib.util.shell.gtdbshutil.symlink"]], "which() (in module gtdblib.util.shell.execute)": [[11, "gtdblib.util.shell.execute.which"]]}})