Search.setIndex({"docnames": ["gtdblib", "gtdblib.exception", "gtdblib.file", "gtdblib.file.itol", "gtdblib.taxon", "gtdblib.taxonomy", "gtdblib.util", "gtdblib.util.bio", "gtdblib.util.shell", "index", "modules"], "filenames": ["gtdblib.rst", "gtdblib.exception.rst", "gtdblib.file.rst", "gtdblib.file.itol.rst", "gtdblib.taxon.rst", "gtdblib.taxonomy.rst", "gtdblib.util.rst", "gtdblib.util.bio.rst", "gtdblib.util.shell.rst", "index.rst", "modules.rst"], "titles": ["gtdblib package", "gtdblib.exception package", "gtdblib.file package", "gtdblib.file.itol package", "gtdblib.taxon package", "gtdblib.taxonomy package", "gtdblib.util package", "gtdblib.util.bio package", "gtdblib.util.shell package", "Welcome to GTDB Library\u2019s documentation!", "gtdblib"], "terms": {"except": [0, 6, 9, 10], "gtdblibexcept": [0, 1, 9], "file": [0, 7, 8, 9, 10], "itol": [0, 2, 9], "submodul": [0, 2, 6, 9, 10], "collaps": [0, 2, 9], "label": [0, 2, 9], "popup": [0, 2, 9], "tree_color": [0, 2, 9], "path": [0, 2, 3, 8, 9], "taxon": [0, 5, 9, 10], "rank": [0, 9, 10], "taxonrank": [0, 4, 9], "class": [0, 2, 3, 4, 5, 9], "domain": [0, 4, 9], "famili": [0, 4, 9], "genu": [0, 4, 9], "order": [0, 4, 7, 9], "phylum": [0, 4, 9], "speci": [0, 4, 9], "valu": [0, 4, 8, 9], "taxonomi": [0, 9, 10], "c": [0, 5, 7, 9], "d": [0, 5, 7, 9], "f": [0, 5, 9], "g": [0, 4, 5, 7, 8, 9], "o": [0, 5, 9], "p": [0, 5, 9], "": [0, 5], "util": [0, 9, 10], "bio": [0, 6, 9], "seq_io": [0, 6, 9], "shell": [0, 6, 9], "checksum": [0, 6, 9], "color": [0, 3, 6, 9], "gtdbshutil": [0, 6, 9], "messag": 1, "str": [1, 3, 4, 7, 8], "base": [1, 2, 3, 4, 5, 7], "all": [1, 2, 3], "gtdb": 1, "lib": 1, "thrown": 1, "thi": [1, 7, 8], "project": 1, "itolcollapsefil": [2, 3, 9], "data": [2, 3, 9], "insert": [2, 3, 7, 9], "write": [2, 3, 7, 9], "itollabelfil": [2, 3, 9], "itolpopupfil": [2, 3, 9], "itoltreecolorsfil": [2, 3, 9], "insert_branch": [2, 3, 9], "insert_clad": [2, 3, 9], "insert_label": [2, 3, 9], "insert_label_background": [2, 3, 9], "insert_rang": [2, 3, 9], "object": [2, 4, 5], "The": [2, 7, 8], "inherit": 2, "from": [2, 7, 8], "list": 3, "node_id": 3, "union": 3, "tupl": [3, 8], "titl": 3, "branch_styl": 3, "scale_factor": 3, "float": [3, 7], "defin": [3, 7], "style": 3, "singl": 3, "branch": 3, "clade": 3, "font_styl": 3, "option": 3, "none": [3, 7], "font": 3, "leaf": 3, "background": 3, "rang": 3, "enum": 4, "A": [4, 7], "taxonom": 4, "e": 4, "etc": 4, "c__": 4, "d__": 4, "f__": 4, "g__": 4, "o__": 4, "p__": 4, "s__": 4, "bioerror": [6, 7, 9], "inputfileerror": [6, 7, 9], "extract_seq": [6, 7, 9], "is_nucleotid": [6, 7, 9], "is_protein": [6, 7, 9], "read": [6, 7, 9], "read_fasta": [6, 7, 9], "read_fasta_seq": [6, 7, 9], "read_fastq": [6, 7, 9], "read_fastq_seq": [6, 7, 9], "read_seq": [6, 7, 9], "seq_length": [6, 7, 9], "write_fasta": [6, 7, 9], "sha256": [6, 8, 9], "sha256_rb": [6, 8, 9], "hex_to_rgb": [6, 8, 9], "rgb_to_hex": [6, 8, 9], "check_dir_exist": [6, 8, 9], "check_file_exist": [6, 8, 9], "get_num_lin": [6, 8, 9], "make_sure_path_exist": [6, 8, 9], "symlink": [6, 8, 9], "msg": 7, "fasta_fil": 7, "seqs_to_extract": 7, "extract": 7, "specif": 7, "sequenc": 7, "fasta": 7, "paramet": [7, 8], "contain": 7, "set": 7, "id": 7, "return": [7, 8], "dict": 7, "seq_id": 7, "seq": 7, "dictionari": 7, "index": 7, "seq_fil": 7, "req_perc": 7, "0": 7, "95": 7, "max_seqs_to_read": 7, "10": [7, 8], "check": [7, 8], "nucleotid": 7, "space": 7, "i": [7, 8], "perform": 7, "look": 7, "charact": 7, "t": 7, "n": 7, "confirm": 7, "compris": 7, "major": 7, "number": [7, 8], "ar": 7, "assum": 7, "primarili": 7, "name": [7, 8], "q": 7, "percentag": 7, "befor": 7, "declar": 7, "being": 7, "int": [7, 8], "maximum": 7, "boolean": 7, "true": [7, 8], "protein": 7, "20": 7, "amino": 7, "acid": 7, "along": 7, "x": 7, "keep_annot": 7, "fals": [7, 8], "determin": [7, 8], "should": 7, "annot": 7, "gener": 7, "function": [7, 8], "intend": 7, "us": 7, "avoid": 7, "have": 7, "larg": 7, "memori": 7, "input": 7, "mai": 7, "gzip": 7, "exampl": 7, "seqio": 7, "print": 7, "string": 7, "iter": 7, "uniqu": 7, "follow": 7, "itself": 7, "fastq_fil": 7, "fastq": 7, "either": 7, "format": [7, 8], "It": 7, "slightli": 7, "more": 7, "effici": 7, "directli": 7, "call": 7, "type": 7, "known": 7, "param": 7, "calcul": [7, 8], "length": 7, "each": 7, "output_fil": 7, "wrap": 7, "80": 7, "If": [7, 8], "output": 7, "ha": 7, "extens": 7, "gz": 7, "compress": 7, "produc": 7, "per": 7, "line": [7, 8], "input_fil": 8, "hash": 8, "binari": 8, "modifi": 8, "hex": 8, "convert": 8, "rgb": 8, "r": 8, "b": 8, "red": 8, "green": 8, "blue": 8, "copi": 8, "archiv": 8, "directori": 8, "tree": 8, "input_dir": 8, "exist": 8, "biolib": 8, "09": 8, "2020": 8, "file_path": 8, "creat": 8, "doe": 8, "wa": 8, "target": 8, "link_nam": 8, "overwrit": 8, "bool": 8, "symbol": 8, "link": 8, "point": 8, "fileexistserror": 8, "rais": 8, "unless": 8, "when": 8, "try": 8, "isadirectoryerror": 8, "gtdblib": 9, "packag": [9, 10], "subpackag": [9, 10], "modul": [9, 10], "content": 10}, "objects": {"": [[0, 0, 0, "-", "gtdblib"]], "gtdblib": [[1, 0, 0, "-", "exception"], [2, 0, 0, "-", "file"], [4, 0, 0, "-", "taxon"], [5, 0, 0, "-", "taxonomy"], [6, 0, 0, "-", "util"]], "gtdblib.exception": [[1, 1, 1, "", "GtdbLibException"]], "gtdblib.file": [[2, 2, 1, "", "File"], [3, 0, 0, "-", "itol"]], "gtdblib.file.File": [[2, 3, 1, "", "path"]], "gtdblib.file.itol": [[3, 0, 0, "-", "collapse"], [3, 0, 0, "-", "label"], [3, 0, 0, "-", "popup"], [3, 0, 0, "-", "tree_colors"]], "gtdblib.file.itol.collapse": [[3, 2, 1, "", "iTolCollapseFile"]], "gtdblib.file.itol.collapse.iTolCollapseFile": [[3, 3, 1, "", "data"], [3, 4, 1, "", "insert"], [3, 4, 1, "", "write"]], "gtdblib.file.itol.label": [[3, 2, 1, "", "iTolLabelFile"]], "gtdblib.file.itol.label.iTolLabelFile": [[3, 3, 1, "", "data"], [3, 4, 1, "", "insert"], [3, 4, 1, "", "write"]], "gtdblib.file.itol.popup": [[3, 2, 1, "", "iTolPopupFile"]], "gtdblib.file.itol.popup.iTolPopupFile": [[3, 3, 1, "", "data"], [3, 4, 1, "", "insert"], [3, 4, 1, "", "write"]], "gtdblib.file.itol.tree_colors": [[3, 2, 1, "", "iTolTreeColorsFile"]], "gtdblib.file.itol.tree_colors.iTolTreeColorsFile": [[3, 3, 1, "", "data"], [3, 4, 1, "", "insert_branch"], [3, 4, 1, "", "insert_clade"], [3, 4, 1, "", "insert_label"], [3, 4, 1, "", "insert_label_background"], [3, 4, 1, "", "insert_range"], [3, 4, 1, "", "write"]], "gtdblib.taxon": [[4, 0, 0, "-", "rank"], [4, 0, 0, "-", "taxon"]], "gtdblib.taxon.rank": [[4, 2, 1, "", "TaxonRank"]], "gtdblib.taxon.rank.TaxonRank": [[4, 3, 1, "", "CLASS"], [4, 3, 1, "", "DOMAIN"], [4, 3, 1, "", "FAMILY"], [4, 3, 1, "", "GENUS"], [4, 3, 1, "", "ORDER"], [4, 3, 1, "", "PHYLUM"], [4, 3, 1, "", "SPECIES"]], "gtdblib.taxon.taxon": [[4, 2, 1, "", "Taxon"]], "gtdblib.taxon.taxon.Taxon": [[4, 3, 1, "", "rank"], [4, 3, 1, "", "value"]], "gtdblib.taxonomy": [[5, 0, 0, "-", "taxonomy"]], "gtdblib.taxonomy.taxonomy": [[5, 2, 1, "", "Taxonomy"]], "gtdblib.taxonomy.taxonomy.Taxonomy": [[5, 3, 1, "", "c"], [5, 3, 1, "", "d"], [5, 3, 1, "", "f"], [5, 3, 1, "", "g"], [5, 3, 1, "", "o"], [5, 3, 1, "", "p"], [5, 3, 1, "", "s"]], "gtdblib.util": [[7, 0, 0, "-", "bio"], [8, 0, 0, "-", "shell"]], "gtdblib.util.bio": [[7, 0, 0, "-", "exceptions"], [7, 0, 0, "-", "seq_io"]], "gtdblib.util.bio.exceptions": [[7, 1, 1, "", "BioError"]], "gtdblib.util.bio.seq_io": [[7, 1, 1, "", "InputFileError"], [7, 5, 1, "", "extract_seqs"], [7, 5, 1, "", "is_nucleotide"], [7, 5, 1, "", "is_protein"], [7, 5, 1, "", "read"], [7, 5, 1, "", "read_fasta"], [7, 5, 1, "", "read_fasta_seq"], [7, 5, 1, "", "read_fastq"], [7, 5, 1, "", "read_fastq_seq"], [7, 5, 1, "", "read_seq"], [7, 5, 1, "", "seq_lengths"], [7, 5, 1, "", "write_fasta"]], "gtdblib.util.shell": [[8, 0, 0, "-", "checksum"], [8, 0, 0, "-", "color"], [8, 0, 0, "-", "gtdbshutil"]], "gtdblib.util.shell.checksum": [[8, 5, 1, "", "sha256"], [8, 5, 1, "", "sha256_rb"]], "gtdblib.util.shell.color": [[8, 5, 1, "", "hex_to_rgb"], [8, 5, 1, "", "rgb_to_hex"]], "gtdblib.util.shell.gtdbshutil": [[8, 5, 1, "", "check_dir_exists"], [8, 5, 1, "", "check_file_exists"], [8, 5, 1, "", "get_num_lines"], [8, 5, 1, "", "make_sure_path_exists"], [8, 5, 1, "", "symlink"]]}, "objtypes": {"0": "py:module", "1": "py:exception", "2": "py:class", "3": "py:attribute", "4": "py:method", "5": "py:function"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "exception", "Python exception"], "2": ["py", "class", "Python class"], "3": ["py", "attribute", "Python attribute"], "4": ["py", "method", "Python method"], "5": ["py", "function", "Python function"]}, "titleterms": {"gtdblib": [0, 1, 2, 3, 4, 5, 6, 7, 8, 10], "packag": [0, 1, 2, 3, 4, 5, 6, 7, 8], "subpackag": [0, 2, 6], "modul": [0, 1, 2, 3, 4, 5, 6, 7, 8], "content": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9], "except": [1, 7], "file": [2, 3], "itol": 3, "submodul": [3, 4, 5, 7, 8], "collaps": 3, "label": 3, "popup": 3, "tree_color": 3, "taxon": 4, "rank": 4, "taxonomi": 5, "util": [6, 7, 8], "bio": 7, "seq_io": 7, "shell": 8, "checksum": 8, "color": 8, "gtdbshutil": 8, "welcom": 9, "gtdb": 9, "librari": 9, "": 9, "document": 9}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx": 57}, "alltitles": {"gtdblib package": [[0, "gtdblib-package"]], "Subpackages": [[0, "subpackages"], [2, "subpackages"], [6, "subpackages"]], "Module contents": [[0, "module-gtdblib"], [1, "module-gtdblib.exception"], [2, "module-gtdblib.file"], [3, "module-gtdblib.file.itol"], [4, "module-gtdblib.taxon"], [5, "module-gtdblib.taxonomy"], [6, "module-gtdblib.util"], [7, "module-gtdblib.util.bio"], [8, "module-gtdblib.util.shell"]], "gtdblib.exception package": [[1, "gtdblib-exception-package"]], "gtdblib.file package": [[2, "gtdblib-file-package"]], "gtdblib.file.itol package": [[3, "gtdblib-file-itol-package"]], "Submodules": [[3, "submodules"], [4, "submodules"], [5, "submodules"], [7, "submodules"], [8, "submodules"]], "gtdblib.file.itol.collapse module": [[3, "module-gtdblib.file.itol.collapse"]], "gtdblib.file.itol.label module": [[3, "module-gtdblib.file.itol.label"]], "gtdblib.file.itol.popup module": [[3, "module-gtdblib.file.itol.popup"]], "gtdblib.file.itol.tree_colors module": [[3, "module-gtdblib.file.itol.tree_colors"]], "gtdblib.taxon package": [[4, "gtdblib-taxon-package"]], "gtdblib.taxon.rank module": [[4, "module-gtdblib.taxon.rank"]], "gtdblib.taxon.taxon module": [[4, "module-gtdblib.taxon.taxon"]], "gtdblib.taxonomy package": [[5, "gtdblib-taxonomy-package"]], "gtdblib.taxonomy.taxonomy module": [[5, "module-gtdblib.taxonomy.taxonomy"]], "gtdblib.util package": [[6, "gtdblib-util-package"]], "gtdblib.util.bio package": [[7, "gtdblib-util-bio-package"]], "gtdblib.util.bio.exceptions module": [[7, "module-gtdblib.util.bio.exceptions"]], "gtdblib.util.bio.seq_io module": [[7, "module-gtdblib.util.bio.seq_io"]], "gtdblib.util.shell package": [[8, "gtdblib-util-shell-package"]], "gtdblib.util.shell.checksum module": [[8, "module-gtdblib.util.shell.checksum"]], "gtdblib.util.shell.color module": [[8, "module-gtdblib.util.shell.color"]], "gtdblib.util.shell.gtdbshutil module": [[8, "module-gtdblib.util.shell.gtdbshutil"]], "Welcome to GTDB Library\u2019s documentation!": [[9, "welcome-to-gtdb-library-s-documentation"]], "Contents:": [[9, null]], "gtdblib": [[10, "gtdblib"]]}, "indexentries": {"gtdblib": [[0, "module-gtdblib"]], "module": [[0, "module-gtdblib"], [1, "module-gtdblib.exception"], [2, "module-gtdblib.file"], [3, "module-gtdblib.file.itol"], [3, "module-gtdblib.file.itol.collapse"], [3, "module-gtdblib.file.itol.label"], [3, "module-gtdblib.file.itol.popup"], [3, "module-gtdblib.file.itol.tree_colors"], [4, "module-gtdblib.taxon"], [4, "module-gtdblib.taxon.rank"], [4, "module-gtdblib.taxon.taxon"], [5, "module-gtdblib.taxonomy"], [5, "module-gtdblib.taxonomy.taxonomy"], [6, "module-gtdblib.util"], [7, "module-gtdblib.util.bio"], [7, "module-gtdblib.util.bio.exceptions"], [7, "module-gtdblib.util.bio.seq_io"], [8, "module-gtdblib.util.shell"], [8, "module-gtdblib.util.shell.checksum"], [8, "module-gtdblib.util.shell.color"], [8, "module-gtdblib.util.shell.gtdbshutil"]], "gtdblibexception": [[1, "gtdblib.exception.GtdbLibException"]], "gtdblib.exception": [[1, "module-gtdblib.exception"]], "file (class in gtdblib.file)": [[2, "gtdblib.file.File"]], "gtdblib.file": [[2, "module-gtdblib.file"]], "path (gtdblib.file.file attribute)": [[2, "gtdblib.file.File.path"]], "data (gtdblib.file.itol.collapse.itolcollapsefile attribute)": [[3, "gtdblib.file.itol.collapse.iTolCollapseFile.data"]], "data (gtdblib.file.itol.label.itollabelfile attribute)": [[3, "gtdblib.file.itol.label.iTolLabelFile.data"]], "data (gtdblib.file.itol.popup.itolpopupfile attribute)": [[3, "gtdblib.file.itol.popup.iTolPopupFile.data"]], "data (gtdblib.file.itol.tree_colors.itoltreecolorsfile attribute)": [[3, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.data"]], "gtdblib.file.itol": [[3, "module-gtdblib.file.itol"]], "gtdblib.file.itol.collapse": [[3, "module-gtdblib.file.itol.collapse"]], "gtdblib.file.itol.label": [[3, "module-gtdblib.file.itol.label"]], "gtdblib.file.itol.popup": [[3, "module-gtdblib.file.itol.popup"]], "gtdblib.file.itol.tree_colors": [[3, "module-gtdblib.file.itol.tree_colors"]], "itolcollapsefile (class in gtdblib.file.itol.collapse)": [[3, "gtdblib.file.itol.collapse.iTolCollapseFile"]], "itollabelfile (class in gtdblib.file.itol.label)": [[3, "gtdblib.file.itol.label.iTolLabelFile"]], "itolpopupfile (class in gtdblib.file.itol.popup)": [[3, "gtdblib.file.itol.popup.iTolPopupFile"]], "itoltreecolorsfile (class in gtdblib.file.itol.tree_colors)": [[3, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile"]], "insert() (gtdblib.file.itol.collapse.itolcollapsefile method)": [[3, "gtdblib.file.itol.collapse.iTolCollapseFile.insert"]], "insert() (gtdblib.file.itol.label.itollabelfile method)": [[3, "gtdblib.file.itol.label.iTolLabelFile.insert"]], "insert() (gtdblib.file.itol.popup.itolpopupfile method)": [[3, "gtdblib.file.itol.popup.iTolPopupFile.insert"]], "insert_branch() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[3, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.insert_branch"]], "insert_clade() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[3, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.insert_clade"]], "insert_label() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[3, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.insert_label"]], "insert_label_background() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[3, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.insert_label_background"]], "insert_range() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[3, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.insert_range"]], "write() (gtdblib.file.itol.collapse.itolcollapsefile method)": [[3, "gtdblib.file.itol.collapse.iTolCollapseFile.write"]], "write() (gtdblib.file.itol.label.itollabelfile method)": [[3, "gtdblib.file.itol.label.iTolLabelFile.write"]], "write() (gtdblib.file.itol.popup.itolpopupfile method)": [[3, "gtdblib.file.itol.popup.iTolPopupFile.write"]], "write() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[3, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.write"]], "class (gtdblib.taxon.rank.taxonrank attribute)": [[4, "gtdblib.taxon.rank.TaxonRank.CLASS"]], "domain (gtdblib.taxon.rank.taxonrank attribute)": [[4, "gtdblib.taxon.rank.TaxonRank.DOMAIN"]], "family (gtdblib.taxon.rank.taxonrank attribute)": [[4, "gtdblib.taxon.rank.TaxonRank.FAMILY"]], "genus (gtdblib.taxon.rank.taxonrank attribute)": [[4, "gtdblib.taxon.rank.TaxonRank.GENUS"]], "order (gtdblib.taxon.rank.taxonrank attribute)": [[4, "gtdblib.taxon.rank.TaxonRank.ORDER"]], "phylum (gtdblib.taxon.rank.taxonrank attribute)": [[4, "gtdblib.taxon.rank.TaxonRank.PHYLUM"]], "species (gtdblib.taxon.rank.taxonrank attribute)": [[4, "gtdblib.taxon.rank.TaxonRank.SPECIES"]], "taxon (class in gtdblib.taxon.taxon)": [[4, "gtdblib.taxon.taxon.Taxon"]], "taxonrank (class in gtdblib.taxon.rank)": [[4, "gtdblib.taxon.rank.TaxonRank"]], "gtdblib.taxon": [[4, "module-gtdblib.taxon"]], "gtdblib.taxon.rank": [[4, "module-gtdblib.taxon.rank"]], "gtdblib.taxon.taxon": [[4, "module-gtdblib.taxon.taxon"]], "rank (gtdblib.taxon.taxon.taxon attribute)": [[4, "gtdblib.taxon.taxon.Taxon.rank"]], "value (gtdblib.taxon.taxon.taxon attribute)": [[4, "gtdblib.taxon.taxon.Taxon.value"]], "taxonomy (class in gtdblib.taxonomy.taxonomy)": [[5, "gtdblib.taxonomy.taxonomy.Taxonomy"]], "c (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[5, "gtdblib.taxonomy.taxonomy.Taxonomy.c"]], "d (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[5, "gtdblib.taxonomy.taxonomy.Taxonomy.d"]], "f (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[5, "gtdblib.taxonomy.taxonomy.Taxonomy.f"]], "g (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[5, "gtdblib.taxonomy.taxonomy.Taxonomy.g"]], "gtdblib.taxonomy": [[5, "module-gtdblib.taxonomy"]], "gtdblib.taxonomy.taxonomy": [[5, "module-gtdblib.taxonomy.taxonomy"]], "o (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[5, "gtdblib.taxonomy.taxonomy.Taxonomy.o"]], "p (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[5, "gtdblib.taxonomy.taxonomy.Taxonomy.p"]], "s (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[5, "gtdblib.taxonomy.taxonomy.Taxonomy.s"]], "gtdblib.util": [[6, "module-gtdblib.util"]], "bioerror": [[7, "gtdblib.util.bio.exceptions.BioError"]], "inputfileerror": [[7, "gtdblib.util.bio.seq_io.InputFileError"]], "extract_seqs() (in module gtdblib.util.bio.seq_io)": [[7, "gtdblib.util.bio.seq_io.extract_seqs"]], "gtdblib.util.bio": [[7, "module-gtdblib.util.bio"]], "gtdblib.util.bio.exceptions": [[7, "module-gtdblib.util.bio.exceptions"]], "gtdblib.util.bio.seq_io": [[7, "module-gtdblib.util.bio.seq_io"]], "is_nucleotide() (in module gtdblib.util.bio.seq_io)": [[7, "gtdblib.util.bio.seq_io.is_nucleotide"]], "is_protein() (in module gtdblib.util.bio.seq_io)": [[7, "gtdblib.util.bio.seq_io.is_protein"]], "read() (in module gtdblib.util.bio.seq_io)": [[7, "gtdblib.util.bio.seq_io.read"]], "read_fasta() (in module gtdblib.util.bio.seq_io)": [[7, "gtdblib.util.bio.seq_io.read_fasta"]], "read_fasta_seq() (in module gtdblib.util.bio.seq_io)": [[7, "gtdblib.util.bio.seq_io.read_fasta_seq"]], "read_fastq() (in module gtdblib.util.bio.seq_io)": [[7, "gtdblib.util.bio.seq_io.read_fastq"]], "read_fastq_seq() (in module gtdblib.util.bio.seq_io)": [[7, "gtdblib.util.bio.seq_io.read_fastq_seq"]], "read_seq() (in module gtdblib.util.bio.seq_io)": [[7, "gtdblib.util.bio.seq_io.read_seq"]], "seq_lengths() (in module gtdblib.util.bio.seq_io)": [[7, "gtdblib.util.bio.seq_io.seq_lengths"]], "write_fasta() (in module gtdblib.util.bio.seq_io)": [[7, "gtdblib.util.bio.seq_io.write_fasta"]], "check_dir_exists() (in module gtdblib.util.shell.gtdbshutil)": [[8, "gtdblib.util.shell.gtdbshutil.check_dir_exists"]], "check_file_exists() (in module gtdblib.util.shell.gtdbshutil)": [[8, "gtdblib.util.shell.gtdbshutil.check_file_exists"]], "get_num_lines() (in module gtdblib.util.shell.gtdbshutil)": [[8, "gtdblib.util.shell.gtdbshutil.get_num_lines"]], "gtdblib.util.shell": [[8, "module-gtdblib.util.shell"]], "gtdblib.util.shell.checksum": [[8, "module-gtdblib.util.shell.checksum"]], "gtdblib.util.shell.color": [[8, "module-gtdblib.util.shell.color"]], "gtdblib.util.shell.gtdbshutil": [[8, "module-gtdblib.util.shell.gtdbshutil"]], "hex_to_rgb() (in module gtdblib.util.shell.color)": [[8, "gtdblib.util.shell.color.hex_to_rgb"]], "make_sure_path_exists() (in module gtdblib.util.shell.gtdbshutil)": [[8, "gtdblib.util.shell.gtdbshutil.make_sure_path_exists"]], "rgb_to_hex() (in module gtdblib.util.shell.color)": [[8, "gtdblib.util.shell.color.rgb_to_hex"]], "sha256() (in module gtdblib.util.shell.checksum)": [[8, "gtdblib.util.shell.checksum.sha256"]], "sha256_rb() (in module gtdblib.util.shell.checksum)": [[8, "gtdblib.util.shell.checksum.sha256_rb"]], "symlink() (in module gtdblib.util.shell.gtdbshutil)": [[8, "gtdblib.util.shell.gtdbshutil.symlink"]]}})