Search.setIndex({"docnames": ["gtdblib", "gtdblib.cli", "gtdblib.exception", "gtdblib.file", "gtdblib.file.itol", "gtdblib.taxon", "gtdblib.taxonomy", "gtdblib.tree", "gtdblib.util", "gtdblib.util.bio", "gtdblib.util.shell", "index", "modules"], "filenames": ["gtdblib.rst", "gtdblib.cli.rst", "gtdblib.exception.rst", "gtdblib.file.rst", "gtdblib.file.itol.rst", "gtdblib.taxon.rst", "gtdblib.taxonomy.rst", "gtdblib.tree.rst", "gtdblib.util.rst", "gtdblib.util.bio.rst", "gtdblib.util.shell.rst", "index.rst", "modules.rst"], "titles": ["gtdblib package", "gtdblib.cli package", "gtdblib.exception package", "gtdblib.file package", "gtdblib.file.itol package", "gtdblib.taxon package", "gtdblib.taxonomy package", "gtdblib.tree package", "gtdblib.util package", "gtdblib.util.bio package", "gtdblib.util.shell package", "Welcome to GTDB Library\u2019s documentation!", "gtdblib"], "terms": {"cli": [0, 11, 12], "submodul": [0, 3, 11, 12], "tree": [0, 4, 6, 9, 10, 11, 12], "bootstrap_merg": [0, 1, 11, 12], "convert_to_canon": [0, 1], "polytomi": [0, 1, 11, 12], "except": [0, 8, 11, 12], "gtdblibexcept": [0, 2], "gtdblibexit": [0, 2], "file": [0, 6, 9, 10, 11, 12], "itol": [0, 3], "collaps": [0, 1, 3, 7], "dataset_color_strip": [0, 3], "dataset_text": [0, 3], "label": [0, 3, 6, 9], "popup": [0, 3], "tree_color": [0, 3], "path": [0, 1, 3, 4, 7, 10], "taxon": [0, 6, 7, 9, 11, 12], "rank": [0, 6, 11, 12], "taxonrank": [0, 5], "class": [0, 3, 4, 5, 6], "domain": [0, 5, 6], "famili": [0, 5, 6], "genu": [0, 5, 6], "order": [0, 5, 6, 9], "phylum": [0, 5, 6], "speci": [0, 5, 6], "valu": [0, 1, 4, 5, 7, 9, 10], "taxonomi": [0, 11, 12], "class_index": [0, 6], "domain_index": [0, 6], "family_index": [0, 6], "genus_index": [0, 6], "order_index": [0, 6], "phylum_index": [0, 6], "rank_index": [0, 6], "rank_label": [0, 6], "rank_prefix": [0, 6], "species_index": [0, 6], "c": [0, 6, 9], "d": [0, 6, 9], "f": [0, 6], "g": [0, 1, 5, 6, 8, 9, 10], "o": [0, 6], "p": [0, 6], "": [0, 6, 8], "fill_trailing_rank": [0, 6], "read_taxonomi": [0, 6], "read_taxonomy_from_tre": [0, 6], "valid": [0, 11, 12], "duplicate_nam": [0, 6], "taxonomic_consist": [0, 6], "validate_species_nam": [0, 6], "validate_taxonomi": [0, 6], "bootstrap_merge_repl": [0, 7], "convert_access": [0, 11, 12], "convert_tree_accessions_to_canon": [0, 7], "get_node_depth": [0, 11, 12], "get_tree_node_depth": [0, 7], "get_node_to_desc_taxa": [0, 11, 12], "get_tree_node_to_desc_taxa": [0, 7], "get_unique_mrca_for_internal_nod": [0, 11, 12], "get_tree_unique_mrca_for_internal_nod": [0, 7], "collapse_polytomi": [0, 7], "collapse_polytomy_tre": [0, 7], "util": [0, 11, 12], "bio": [0, 8], "access": [0, 1, 6, 7, 8], "newick": [0, 6, 8], "seq_io": [0, 8], "shell": [0, 8], "color": [0, 4, 11, 12], "execut": [0, 8], "filemgmt": [0, 8], "gtdbshutil": [0, 8], "rgb_to_hex": [0, 8, 10], "type": [0, 1, 9, 11, 12], "is_float": [0, 8], "pathlib": 1, "typer": 1, "model": 1, "argumentinfo": 1, "object": [1, 3, 5, 6, 7, 8], "output": [1, 9, 10], "replic": 1, "list": [1, 4, 6, 10], "cpu": [1, 7], "int": [1, 4, 7, 8, 9, 10], "optioninfo": 1, "calcul": [1, 7, 9, 10], "non": [1, 7], "parametr": [1, 7], "bootstrap": [1, 7], "us": [1, 6, 7, 9], "multipl": 1, "convert": [1, 6, 7, 8, 10], "canon": [1, 6, 7, 9], "form": [1, 6, 7, 9], "e": [1, 4, 5, 6, 7, 9], "gca_123456789": 1, "1": [1, 4, 6, 7, 9, 10], "g123456789": 1, "support": [1, 7, 9], "float": [1, 4, 7, 8, 9], "node": [1, 4, 7], "low": [1, 7], "messag": [2, 6], "str": [2, 4, 5, 6, 7, 8, 9, 10], "base": [2, 3, 4, 5, 6, 9, 10], "all": [2, 3, 4, 6, 7, 10], "gtdb": 2, "lib": 2, "thrown": 2, "thi": [2, 6, 7, 9, 10], "project": 2, "rais": [2, 10], "when": [2, 10], "ha": [2, 9], "been": 2, "ask": 2, "quietli": 2, "exit": [2, 10], "itolcollapsefil": [3, 4], "data": [3, 4], "insert": [3, 4, 9], "write": [3, 4, 7, 9], "itoldatasetcolorstripfil": [3, 4], "itoldatasettextfil": [3, 4], "itollabelfil": [3, 4], "itolpopupfil": [3, 4], "itoltreecolorsfil": [3, 4], "insert_branch": [3, 4], "insert_clad": [3, 4], "insert_label": [3, 4], "insert_label_background": [3, 4], "insert_rang": [3, 4], "The": [3, 7, 9, 10], "inherit": 3, "from": [3, 6, 7, 9, 10], "node_id": 4, "color_branch": 4, "bool": [4, 6, 8, 9, 10], "fals": [4, 6, 9, 10], "strip_width": 4, "25": 4, "margin": 4, "0": [4, 6, 9], "border_width": 4, "border_color": 4, "0000ff": 4, "complete_bord": 4, "show_intern": 4, "show_strip_label": 4, "true": [4, 6, 9, 10], "strip_label_posit": 4, "liter": 4, "top": 4, "center": 4, "bottom": 4, "strip_label_size_factor": 4, "strip_label_rot": 4, "strip_label_shift": 4, "strip_label_color": 4, "000000": 4, "strip_label_outlin": 4, "5": [4, 6], "show_label": 4, "size_factor": 4, "label_rot": 4, "label_shift": 4, "labels_below": 4, "vertical_shift": 4, "straight_label": 4, "align_to_tre": 4, "external_label_shift": 4, "label_size_factor": 4, "posit": 4, "style": [4, 6], "normal": 4, "bold": 4, "ital": 4, "rotat": 4, "defin": [4, 9], "text": 4, "extern": [4, 10], "number": [4, 6, 7, 9, 10], "between": 4, "intern": [4, 6, 7], "specifi": [4, 6], "along": [4, 9], "branch": 4, "exampl": [4, 6, 9], "i": [4, 6, 7, 9, 10], "exactli": 4, "start": [4, 6], "middl": 4, "end": 4, "titl": 4, "branch_styl": 4, "scale_factor": 4, "singl": 4, "clade": 4, "font_styl": 4, "option": [4, 9], "none": [4, 6, 9, 10], "font": 4, "leaf": 4, "background": 4, "rang": 4, "enum": 5, "A": [5, 6, 7, 9], "taxonom": [5, 6], "etc": 5, "c__": [5, 6], "d__": [5, 6], "f__": [5, 6], "g__": [5, 6], "o__": [5, 6], "p__": [5, 6], "s__": [5, 6], "2": 6, "4": 6, "3": 6, "6": 6, "taxa": [6, 7], "fill": 6, "miss": 6, "trail": 6, "string": [6, 8, 9, 10], "function": [6, 9, 10], "assum": [6, 9], "provid": [6, 9], "7": 6, "ani": [6, 8, 10], "ar": [6, 7, 9, 10], "d__bacteria": 6, "d__firmicut": 6, "paramet": [6, 7, 8, 9, 10], "return": [6, 7, 9, 10], "taxonomy_fil": 6, "use_canonical_gid": 6, "dict": [6, 7, 9], "read": [6, 8, 9], "greengen": 6, "method": [6, 9], "gener": [6, 9], "can": [6, 8, 10], "doe": [6, 10], "strictli": 6, "requir": 6, "seven": 6, "present": 6, "expect": 6, "format": [6, 9, 10], "where": 6, "typic": 6, "have": [6, 7, 9], "indic": 6, "assign": 6, "flag": 6, "should": [6, 9], "map": 6, "warn": 6, "obtain": 6, "each": [6, 7, 9], "extant": 6, "extran": 6, "classifi": 6, "theseven": 6, "filenam": 6, "dendropi": [6, 7], "issu": 6, "log": 6, "extent": 6, "check_speci": 6, "identifi": [6, 7], "duplic": 6, "name": [6, 9, 10], "unique_id": 6, "index": [6, 9, 11], "uniqu": [6, 7, 9], "id": [6, 9], "lineag": 6, "report_error": 6, "determin": [6, 9], "consist": 6, "classif": 6, "boolean": [6, 9], "error": [6, 10], "written": 6, "screen": 6, "parent": 6, "inconsist": 6, "species_nam": 6, "require_ful": 6, "require_prefix": 6, "tupl": [6, 7, 9, 10], "full": 6, "binomi": 6, "includ": 6, "specif": [6, 9, 10], "epithet": 6, "escherichia": 6, "coli": 6, "two": 6, "separ": 6, "space": [6, 9], "must": 6, "prefix": 6, "otherwis": 6, "reason": 6, "fail": 6, "check_prefix": 6, "check_rank": 6, "check_hierarchi": 6, "check_group_nam": 6, "check_duplicate_nam": 6, "check_capit": 6, "check": [6, 8, 9, 10], "strict": 6, "hierarchi": 6, "implement": [6, 10], "workflow": 6, "set": [6, 7, 9], "It": [6, 9], "common": 6, "configur": 6, "input": [6, 7, 9], "presenc": 6, "group": [6, 7], "invalid": 6, "charact": [6, 9], "taxon_id": 6, "child_taxon_id": 6, "more": [6, 9], "input_tre": 7, "output_tre": 7, "replicate_tre": 7, "collect": 7, "monophylet": 7, "transpos": 7, "them": 7, "merg": 7, "process": [7, 10], "input_path": 7, "output_path": 7, "depth": 7, "dictionari": [7, 9], "frozenset": 7, "get": [7, 9], "descend": 7, "mrca": 7, "alwai": 7, "yield": 7, "same": [7, 9], "path_in": 7, "path_out": 7, "less": 7, "than": 7, "work": 7, "canonical_gid": [8, 9], "is_same_accn_vers": [8, 9], "bioerror": [8, 9], "create_label": [8, 9], "parse_label": [8, 9], "inputfileerror": [8, 9], "extract_seq": [8, 9], "is_nucleotid": [8, 9], "is_protein": [8, 9], "read_fasta": [8, 9], "read_fasta_seq": [8, 9], "read_fastq": [8, 9], "read_fastq_seq": [8, 9], "read_seq": [8, 9], "seq_length": [8, 9], "write_fasta": [8, 9], "hex_to_rgb": [8, 10], "check_depend": [8, 10], "check_on_path": [8, 10], "is_execut": [8, 10], "run": [8, 10], "which": [8, 9, 10], "matching_bracket": [8, 10], "select_delimit": [8, 10], "sha256": [8, 10], "sha256_rb": [8, 10], "check_dir_exist": [8, 10], "check_file_exist": [8, 10], "get_num_lin": [8, 10], "make_sure_path_exist": [8, 10], "symlink": [8, 10], "r": [8, 10], "b": [8, 10], "gid": 9, "ncbi": 9, "genom": 9, "g005435135": 9, "gcf_005435135": 9, "1_asm543513v1_genom": 9, "rs_gcf_005435135": 9, "gb_gca_005435135": 9, "acces": 9, "conver": 9, "accn1": 9, "accn2": 9, "version": 9, "suffix": 9, "proceed": 9, "period": 9, "first": 9, "second": 9, "els": [9, 10], "msg": 9, "auxiliary_info": 9, "creat": [9, 10], "auxiliari": 9, "inform": 9, "pars": 9, "mai": 9, "contain": 9, "fasta_fil": 9, "seqs_to_extract": 9, "extract": 9, "sequenc": 9, "fasta": 9, "seq_id": 9, "seq": 9, "seq_fil": 9, "req_perc": 9, "95": 9, "max_seqs_to_read": 9, "10": [9, 10], "nucleotid": 9, "perform": 9, "look": 9, "t": [9, 10], "n": 9, "confirm": 9, "compris": 9, "major": 9, "primarili": 9, "q": 9, "percentag": 9, "befor": 9, "declar": 9, "being": 9, "maximum": 9, "protein": 9, "20": 9, "amino": 9, "acid": 9, "x": 9, "keep_annot": 9, "annot": 9, "intend": 9, "avoid": 9, "larg": 9, "memori": 9, "gzip": 9, "seqio": 9, "print": 9, "iter": 9, "follow": 9, "itself": 9, "fastq_fil": 9, "fastq": 9, "either": 9, "slightli": 9, "effici": 9, "directli": 9, "call": 9, "known": 9, "param": [9, 10], "length": 9, "output_fil": 9, "wrap": 9, "80": 9, "If": [9, 10], "extens": 9, "gz": 9, "compress": 9, "produc": 9, "per": 9, "line": [9, 10], "modifi": 10, "hex": 10, "rgb": 10, "red": 10, "green": 10, "blue": 10, "program": 10, "exit_on_fail": 10, "system": 10, "code": 10, "fpath": 10, "cmd": 10, "command": 10, "stdout": 10, "python": 10, "linux": 10, "http": 10, "stackoverflow": 10, "com": 10, "question": 10, "377017": 10, "test": 10, "exist": 10, "isn": 10, "split": 10, "match": 10, "bracket": 10, "substr": 10, "so": 10, "we": 10, "nest": 10, "input_fil": 10, "select": 10, "delimit": 10, "csv": 10, "tsv": 10, "hash": 10, "binari": 10, "copi": 10, "archiv": 10, "directori": 10, "input_dir": 10, "biolib": 10, "09": 10, "2020": 10, "file_path": 10, "wa": 10, "target": 10, "link_nam": 10, "overwrit": 10, "symbol": 10, "link": 10, "point": 10, "fileexistserror": 10, "unless": 10, "try": 10, "isadirectoryerror": 10, "gtdblib": 11, "packag": [11, 12], "subpackag": [11, 12], "modul": [11, 12], "search": 11, "page": 11, "content": 12}, "objects": {"": [[0, 0, 0, "-", "gtdblib"]], "gtdblib": [[1, 0, 0, "-", "cli"], [2, 0, 0, "-", "exception"], [3, 0, 0, "-", "file"], [5, 0, 0, "-", "taxon"], [6, 0, 0, "-", "taxonomy"], [7, 0, 0, "-", "tree"], [8, 0, 0, "-", "util"]], "gtdblib.cli": [[1, 0, 0, "-", "tree"]], "gtdblib.cli.tree": [[1, 1, 1, "", "bootstrap_merge"], [1, 1, 1, "", "convert_to_canonical"], [1, 1, 1, "", "polytomy"]], "gtdblib.exception": [[2, 2, 1, "", "GtdbLibException"], [2, 2, 1, "", "GtdbLibExit"]], "gtdblib.file": [[3, 3, 1, "", "File"], [4, 0, 0, "-", "itol"]], "gtdblib.file.File": [[3, 4, 1, "", "path"]], "gtdblib.file.itol": [[4, 0, 0, "-", "collapse"], [4, 0, 0, "-", "dataset_color_strip"], [4, 0, 0, "-", "dataset_text"], [4, 0, 0, "-", "label"], [4, 0, 0, "-", "popup"], [4, 0, 0, "-", "tree_colors"]], "gtdblib.file.itol.collapse": [[4, 3, 1, "", "iTolCollapseFile"]], "gtdblib.file.itol.collapse.iTolCollapseFile": [[4, 4, 1, "", "data"], [4, 5, 1, "", "insert"], [4, 5, 1, "", "write"]], "gtdblib.file.itol.dataset_color_strip": [[4, 3, 1, "", "iTolDatasetColorStripFile"]], "gtdblib.file.itol.dataset_color_strip.iTolDatasetColorStripFile": [[4, 4, 1, "", "data"], [4, 5, 1, "", "insert"], [4, 5, 1, "", "write"]], "gtdblib.file.itol.dataset_text": [[4, 3, 1, "", "iTolDatasetTextFile"]], "gtdblib.file.itol.dataset_text.iTolDatasetTextFile": [[4, 4, 1, "", "data"], [4, 5, 1, "", "insert"], [4, 5, 1, "", "write"]], "gtdblib.file.itol.label": [[4, 3, 1, "", "iTolLabelFile"]], "gtdblib.file.itol.label.iTolLabelFile": [[4, 4, 1, "", "data"], [4, 5, 1, "", "insert"], [4, 5, 1, "", "write"]], "gtdblib.file.itol.popup": [[4, 3, 1, "", "iTolPopupFile"]], "gtdblib.file.itol.popup.iTolPopupFile": [[4, 4, 1, "", "data"], [4, 5, 1, "", "insert"], [4, 5, 1, "", "write"]], "gtdblib.file.itol.tree_colors": [[4, 3, 1, "", "iTolTreeColorsFile"]], "gtdblib.file.itol.tree_colors.iTolTreeColorsFile": [[4, 4, 1, "", "data"], [4, 5, 1, "", "insert_branch"], [4, 5, 1, "", "insert_clade"], [4, 5, 1, "", "insert_label"], [4, 5, 1, "", "insert_label_background"], [4, 5, 1, "", "insert_range"], [4, 5, 1, "", "write"]], "gtdblib.taxon": [[5, 0, 0, "-", "rank"], [5, 0, 0, "-", "taxon"]], "gtdblib.taxon.rank": [[5, 3, 1, "", "TaxonRank"]], "gtdblib.taxon.rank.TaxonRank": [[5, 4, 1, "", "CLASS"], [5, 4, 1, "", "DOMAIN"], [5, 4, 1, "", "FAMILY"], [5, 4, 1, "", "GENUS"], [5, 4, 1, "", "ORDER"], [5, 4, 1, "", "PHYLUM"], [5, 4, 1, "", "SPECIES"]], "gtdblib.taxon.taxon": [[5, 3, 1, "", "Taxon"]], "gtdblib.taxon.taxon.Taxon": [[5, 4, 1, "", "rank"], [5, 4, 1, "", "value"]], "gtdblib.taxonomy": [[6, 0, 0, "-", "taxonomy"], [6, 0, 0, "-", "validation"]], "gtdblib.taxonomy.taxonomy": [[6, 3, 1, "", "Taxonomy"], [6, 1, 1, "", "fill_trailing_ranks"], [6, 1, 1, "", "read_taxonomy"], [6, 1, 1, "", "read_taxonomy_from_tree"]], "gtdblib.taxonomy.taxonomy.Taxonomy": [[6, 4, 1, "", "CLASS_INDEX"], [6, 4, 1, "", "DOMAIN_INDEX"], [6, 4, 1, "", "FAMILY_INDEX"], [6, 4, 1, "", "GENUS_INDEX"], [6, 4, 1, "", "ORDER_INDEX"], [6, 4, 1, "", "PHYLUM_INDEX"], [6, 4, 1, "", "RANK_INDEX"], [6, 4, 1, "", "RANK_LABELS"], [6, 4, 1, "", "RANK_PREFIXES"], [6, 4, 1, "", "SPECIES_INDEX"], [6, 4, 1, "", "c"], [6, 4, 1, "", "d"], [6, 4, 1, "", "f"], [6, 4, 1, "", "g"], [6, 4, 1, "", "o"], [6, 4, 1, "", "p"], [6, 4, 1, "", "s"]], "gtdblib.taxonomy.validation": [[6, 1, 1, "", "duplicate_names"], [6, 1, 1, "", "taxonomic_consistency"], [6, 1, 1, "", "validate_species_name"], [6, 1, 1, "", "validate_taxonomy"]], "gtdblib.tree": [[7, 0, 0, "-", "bootstrap_merge"], [7, 0, 0, "-", "convert_accession"], [7, 0, 0, "-", "get_node_depth"], [7, 0, 0, "-", "get_node_to_desc_taxa"], [7, 0, 0, "-", "get_unique_mrca_for_internal_nodes"], [7, 0, 0, "-", "polytomy"]], "gtdblib.tree.bootstrap_merge": [[7, 1, 1, "", "bootstrap_merge_replicates"]], "gtdblib.tree.convert_accession": [[7, 1, 1, "", "convert_tree_accessions_to_canonical"]], "gtdblib.tree.get_node_depth": [[7, 1, 1, "", "get_tree_node_depth"]], "gtdblib.tree.get_node_to_desc_taxa": [[7, 1, 1, "", "get_tree_node_to_desc_taxa"]], "gtdblib.tree.get_unique_mrca_for_internal_nodes": [[7, 1, 1, "", "get_tree_unique_mrca_for_internal_nodes"]], "gtdblib.tree.polytomy": [[7, 1, 1, "", "collapse_polytomy"], [7, 1, 1, "", "collapse_polytomy_tree"]], "gtdblib.util": [[9, 0, 0, "-", "bio"], [8, 0, 0, "-", "color"], [10, 0, 0, "-", "shell"], [8, 0, 0, "-", "types"]], "gtdblib.util.bio": [[9, 0, 0, "-", "accession"], [9, 0, 0, "-", "exceptions"], [9, 0, 0, "-", "newick"], [9, 0, 0, "-", "seq_io"]], "gtdblib.util.bio.accession": [[9, 1, 1, "", "canonical_gid"], [9, 1, 1, "", "is_same_accn_version"]], "gtdblib.util.bio.exceptions": [[9, 2, 1, "", "BioError"]], "gtdblib.util.bio.newick": [[9, 1, 1, "", "create_label"], [9, 1, 1, "", "parse_label"]], "gtdblib.util.bio.seq_io": [[9, 2, 1, "", "InputFileError"], [9, 1, 1, "", "extract_seqs"], [9, 1, 1, "", "is_nucleotide"], [9, 1, 1, "", "is_protein"], [9, 1, 1, "", "read"], [9, 1, 1, "", "read_fasta"], [9, 1, 1, "", "read_fasta_seq"], [9, 1, 1, "", "read_fastq"], [9, 1, 1, "", "read_fastq_seq"], [9, 1, 1, "", "read_seq"], [9, 1, 1, "", "seq_lengths"], [9, 1, 1, "", "write_fasta"]], "gtdblib.util.color": [[8, 1, 1, "", "rgb_to_hex"]], "gtdblib.util.shell": [[10, 0, 0, "-", "color"], [10, 0, 0, "-", "execute"], [10, 0, 0, "-", "filemgmt"], [10, 0, 0, "-", "gtdbshutil"]], "gtdblib.util.shell.color": [[10, 1, 1, "", "hex_to_rgb"], [10, 1, 1, "", "rgb_to_hex"]], "gtdblib.util.shell.execute": [[10, 1, 1, "", "check_dependencies"], [10, 1, 1, "", "check_on_path"], [10, 1, 1, "", "is_executable"], [10, 1, 1, "", "run"], [10, 1, 1, "", "which"]], "gtdblib.util.shell.filemgmt": [[10, 1, 1, "", "matching_brackets"], [10, 1, 1, "", "select_delimiter"], [10, 1, 1, "", "sha256"], [10, 1, 1, "", "sha256_rb"]], "gtdblib.util.shell.gtdbshutil": [[10, 1, 1, "", "check_dir_exists"], [10, 1, 1, "", "check_file_exists"], [10, 1, 1, "", "get_num_lines"], [10, 1, 1, "", "make_sure_path_exists"], [10, 1, 1, "", "symlink"]], "gtdblib.util.types": [[8, 1, 1, "", "is_float"]]}, "objtypes": {"0": "py:module", "1": "py:function", "2": "py:exception", "3": "py:class", "4": "py:attribute", "5": "py:method"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "function", "Python function"], "2": ["py", "exception", "Python exception"], "3": ["py", "class", "Python class"], "4": ["py", "attribute", "Python attribute"], "5": ["py", "method", "Python method"]}, "titleterms": {"gtdblib": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 12], "packag": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10], "subpackag": [0, 3, 8], "modul": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10], "content": [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11], "cli": 1, "submodul": [1, 4, 5, 6, 7, 8, 9, 10], "tree": [1, 7], "except": [2, 9], "file": [3, 4], "itol": 4, "collaps": 4, "dataset_color_strip": 4, "dataset_text": 4, "label": 4, "popup": 4, "tree_color": 4, "taxon": 5, "rank": 5, "taxonomi": 6, "valid": 6, "bootstrap_merg": 7, "convert_access": 7, "get_node_depth": 7, "get_node_to_desc_taxa": 7, "get_unique_mrca_for_internal_nod": 7, "polytomi": 7, "util": [8, 9, 10], "color": [8, 10], "type": 8, "bio": 9, "access": 9, "newick": 9, "seq_io": 9, "shell": 10, "execut": 10, "filemgmt": 10, "gtdbshutil": 10, "welcom": 11, "gtdb": 11, "librari": 11, "": 11, "document": 11, "indic": 11, "tabl": 11}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx": 57}, "alltitles": {"gtdblib package": [[0, "gtdblib-package"]], "Subpackages": [[0, "subpackages"], [3, "subpackages"], [8, "subpackages"]], "Module contents": [[0, "module-gtdblib"], [1, "module-gtdblib.cli"], [2, "module-gtdblib.exception"], [3, "module-gtdblib.file"], [4, "module-gtdblib.file.itol"], [5, "module-gtdblib.taxon"], [6, "module-gtdblib.taxonomy"], [7, "module-gtdblib.tree"], [8, "module-gtdblib.util"], [9, "module-gtdblib.util.bio"], [10, "module-gtdblib.util.shell"]], "gtdblib.cli package": [[1, "gtdblib-cli-package"]], "Submodules": [[1, "submodules"], [4, "submodules"], [5, "submodules"], [6, "submodules"], [7, "submodules"], [8, "submodules"], [9, "submodules"], [10, "submodules"]], "gtdblib.cli.tree module": [[1, "module-gtdblib.cli.tree"]], "gtdblib.exception package": [[2, "gtdblib-exception-package"]], "gtdblib.file package": [[3, "gtdblib-file-package"]], "gtdblib.file.itol package": [[4, "gtdblib-file-itol-package"]], "gtdblib.file.itol.collapse module": [[4, "module-gtdblib.file.itol.collapse"]], "gtdblib.file.itol.dataset_color_strip module": [[4, "module-gtdblib.file.itol.dataset_color_strip"]], "gtdblib.file.itol.dataset_text module": [[4, "module-gtdblib.file.itol.dataset_text"]], "gtdblib.file.itol.label module": [[4, "module-gtdblib.file.itol.label"]], "gtdblib.file.itol.popup module": [[4, "module-gtdblib.file.itol.popup"]], "gtdblib.file.itol.tree_colors module": [[4, "module-gtdblib.file.itol.tree_colors"]], "gtdblib.taxon package": [[5, "gtdblib-taxon-package"]], "gtdblib.taxon.rank module": [[5, "module-gtdblib.taxon.rank"]], "gtdblib.taxon.taxon module": [[5, "module-gtdblib.taxon.taxon"]], "gtdblib.taxonomy package": [[6, "gtdblib-taxonomy-package"]], "gtdblib.taxonomy.taxonomy module": [[6, "module-gtdblib.taxonomy.taxonomy"]], "gtdblib.taxonomy.validation module": [[6, "module-gtdblib.taxonomy.validation"]], "gtdblib.tree package": [[7, "gtdblib-tree-package"]], "gtdblib.tree.bootstrap_merge module": [[7, "module-gtdblib.tree.bootstrap_merge"]], "gtdblib.tree.convert_accession module": [[7, "module-gtdblib.tree.convert_accession"]], "gtdblib.tree.get_node_depth module": [[7, "module-gtdblib.tree.get_node_depth"]], "gtdblib.tree.get_node_to_desc_taxa module": [[7, "module-gtdblib.tree.get_node_to_desc_taxa"]], "gtdblib.tree.get_unique_mrca_for_internal_nodes module": [[7, "module-gtdblib.tree.get_unique_mrca_for_internal_nodes"]], "gtdblib.tree.polytomy module": [[7, "module-gtdblib.tree.polytomy"]], "gtdblib.util package": [[8, "gtdblib-util-package"]], "gtdblib.util.color module": [[8, "module-gtdblib.util.color"]], "gtdblib.util.types module": [[8, "module-gtdblib.util.types"]], "gtdblib.util.bio package": [[9, "gtdblib-util-bio-package"]], "gtdblib.util.bio.accession module": [[9, "module-gtdblib.util.bio.accession"]], "gtdblib.util.bio.exceptions module": [[9, "module-gtdblib.util.bio.exceptions"]], "gtdblib.util.bio.newick module": [[9, "module-gtdblib.util.bio.newick"]], "gtdblib.util.bio.seq_io module": [[9, "module-gtdblib.util.bio.seq_io"]], "gtdblib.util.shell package": [[10, "gtdblib-util-shell-package"]], "gtdblib.util.shell.color module": [[10, "module-gtdblib.util.shell.color"]], "gtdblib.util.shell.execute module": [[10, "module-gtdblib.util.shell.execute"]], "gtdblib.util.shell.filemgmt module": [[10, "module-gtdblib.util.shell.filemgmt"]], "gtdblib.util.shell.gtdbshutil module": [[10, "module-gtdblib.util.shell.gtdbshutil"]], "Welcome to GTDB Library\u2019s documentation!": [[11, "welcome-to-gtdb-library-s-documentation"]], "Contents:": [[11, null]], "Indices and tables": [[11, "indices-and-tables"]], "gtdblib": [[12, "gtdblib"]]}, "indexentries": {"gtdblib": [[0, "module-gtdblib"]], "module": [[0, "module-gtdblib"], [1, "module-gtdblib.cli"], [1, "module-gtdblib.cli.tree"], [2, "module-gtdblib.exception"], [3, "module-gtdblib.file"], [4, "module-gtdblib.file.itol"], [4, "module-gtdblib.file.itol.collapse"], [4, "module-gtdblib.file.itol.dataset_color_strip"], [4, "module-gtdblib.file.itol.dataset_text"], [4, "module-gtdblib.file.itol.label"], [4, "module-gtdblib.file.itol.popup"], [4, "module-gtdblib.file.itol.tree_colors"], [5, "module-gtdblib.taxon"], [5, "module-gtdblib.taxon.rank"], [5, "module-gtdblib.taxon.taxon"], [6, "module-gtdblib.taxonomy"], [6, "module-gtdblib.taxonomy.taxonomy"], [6, "module-gtdblib.taxonomy.validation"], [7, "module-gtdblib.tree"], [7, "module-gtdblib.tree.bootstrap_merge"], [7, "module-gtdblib.tree.convert_accession"], [7, "module-gtdblib.tree.get_node_depth"], [7, "module-gtdblib.tree.get_node_to_desc_taxa"], [7, "module-gtdblib.tree.get_unique_mrca_for_internal_nodes"], [7, "module-gtdblib.tree.polytomy"], [8, "module-gtdblib.util"], [8, "module-gtdblib.util.color"], [8, "module-gtdblib.util.types"], [9, "module-gtdblib.util.bio"], [9, "module-gtdblib.util.bio.accession"], [9, "module-gtdblib.util.bio.exceptions"], [9, "module-gtdblib.util.bio.newick"], [9, "module-gtdblib.util.bio.seq_io"], [10, "module-gtdblib.util.shell"], [10, "module-gtdblib.util.shell.color"], [10, "module-gtdblib.util.shell.execute"], [10, "module-gtdblib.util.shell.filemgmt"], [10, "module-gtdblib.util.shell.gtdbshutil"]], "bootstrap_merge() (in module gtdblib.cli.tree)": [[1, "gtdblib.cli.tree.bootstrap_merge"]], "convert_to_canonical() (in module gtdblib.cli.tree)": [[1, "gtdblib.cli.tree.convert_to_canonical"]], "gtdblib.cli": [[1, "module-gtdblib.cli"]], "gtdblib.cli.tree": [[1, "module-gtdblib.cli.tree"]], "polytomy() (in module gtdblib.cli.tree)": [[1, "gtdblib.cli.tree.polytomy"]], "gtdblibexception": [[2, "gtdblib.exception.GtdbLibException"]], "gtdblibexit": [[2, "gtdblib.exception.GtdbLibExit"]], "gtdblib.exception": [[2, "module-gtdblib.exception"]], "file (class in gtdblib.file)": [[3, "gtdblib.file.File"]], "gtdblib.file": [[3, "module-gtdblib.file"]], "path (gtdblib.file.file attribute)": [[3, "gtdblib.file.File.path"]], "data (gtdblib.file.itol.collapse.itolcollapsefile attribute)": [[4, "gtdblib.file.itol.collapse.iTolCollapseFile.data"]], "data (gtdblib.file.itol.dataset_color_strip.itoldatasetcolorstripfile attribute)": [[4, "gtdblib.file.itol.dataset_color_strip.iTolDatasetColorStripFile.data"]], "data (gtdblib.file.itol.dataset_text.itoldatasettextfile attribute)": [[4, "gtdblib.file.itol.dataset_text.iTolDatasetTextFile.data"]], "data (gtdblib.file.itol.label.itollabelfile attribute)": [[4, "gtdblib.file.itol.label.iTolLabelFile.data"]], "data (gtdblib.file.itol.popup.itolpopupfile attribute)": [[4, "gtdblib.file.itol.popup.iTolPopupFile.data"]], "data (gtdblib.file.itol.tree_colors.itoltreecolorsfile attribute)": [[4, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.data"]], "gtdblib.file.itol": [[4, "module-gtdblib.file.itol"]], "gtdblib.file.itol.collapse": [[4, "module-gtdblib.file.itol.collapse"]], "gtdblib.file.itol.dataset_color_strip": [[4, "module-gtdblib.file.itol.dataset_color_strip"]], "gtdblib.file.itol.dataset_text": [[4, "module-gtdblib.file.itol.dataset_text"]], "gtdblib.file.itol.label": [[4, "module-gtdblib.file.itol.label"]], "gtdblib.file.itol.popup": [[4, "module-gtdblib.file.itol.popup"]], "gtdblib.file.itol.tree_colors": [[4, "module-gtdblib.file.itol.tree_colors"]], "itolcollapsefile (class in gtdblib.file.itol.collapse)": [[4, "gtdblib.file.itol.collapse.iTolCollapseFile"]], "itoldatasetcolorstripfile (class in gtdblib.file.itol.dataset_color_strip)": [[4, "gtdblib.file.itol.dataset_color_strip.iTolDatasetColorStripFile"]], "itoldatasettextfile (class in gtdblib.file.itol.dataset_text)": [[4, "gtdblib.file.itol.dataset_text.iTolDatasetTextFile"]], "itollabelfile (class in gtdblib.file.itol.label)": [[4, "gtdblib.file.itol.label.iTolLabelFile"]], "itolpopupfile (class in gtdblib.file.itol.popup)": [[4, "gtdblib.file.itol.popup.iTolPopupFile"]], "itoltreecolorsfile (class in gtdblib.file.itol.tree_colors)": [[4, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile"]], "insert() (gtdblib.file.itol.collapse.itolcollapsefile method)": [[4, "gtdblib.file.itol.collapse.iTolCollapseFile.insert"]], "insert() (gtdblib.file.itol.dataset_color_strip.itoldatasetcolorstripfile method)": [[4, "gtdblib.file.itol.dataset_color_strip.iTolDatasetColorStripFile.insert"]], "insert() (gtdblib.file.itol.dataset_text.itoldatasettextfile method)": [[4, "gtdblib.file.itol.dataset_text.iTolDatasetTextFile.insert"]], "insert() (gtdblib.file.itol.label.itollabelfile method)": [[4, "gtdblib.file.itol.label.iTolLabelFile.insert"]], "insert() (gtdblib.file.itol.popup.itolpopupfile method)": [[4, "gtdblib.file.itol.popup.iTolPopupFile.insert"]], "insert_branch() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[4, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.insert_branch"]], "insert_clade() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[4, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.insert_clade"]], "insert_label() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[4, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.insert_label"]], "insert_label_background() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[4, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.insert_label_background"]], "insert_range() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[4, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.insert_range"]], "write() (gtdblib.file.itol.collapse.itolcollapsefile method)": [[4, "gtdblib.file.itol.collapse.iTolCollapseFile.write"]], "write() (gtdblib.file.itol.dataset_color_strip.itoldatasetcolorstripfile method)": [[4, "gtdblib.file.itol.dataset_color_strip.iTolDatasetColorStripFile.write"]], "write() (gtdblib.file.itol.dataset_text.itoldatasettextfile method)": [[4, "gtdblib.file.itol.dataset_text.iTolDatasetTextFile.write"]], "write() (gtdblib.file.itol.label.itollabelfile method)": [[4, "gtdblib.file.itol.label.iTolLabelFile.write"]], "write() (gtdblib.file.itol.popup.itolpopupfile method)": [[4, "gtdblib.file.itol.popup.iTolPopupFile.write"]], "write() (gtdblib.file.itol.tree_colors.itoltreecolorsfile method)": [[4, "gtdblib.file.itol.tree_colors.iTolTreeColorsFile.write"]], "class (gtdblib.taxon.rank.taxonrank attribute)": [[5, "gtdblib.taxon.rank.TaxonRank.CLASS"]], "domain (gtdblib.taxon.rank.taxonrank attribute)": [[5, "gtdblib.taxon.rank.TaxonRank.DOMAIN"]], "family (gtdblib.taxon.rank.taxonrank attribute)": [[5, "gtdblib.taxon.rank.TaxonRank.FAMILY"]], "genus (gtdblib.taxon.rank.taxonrank attribute)": [[5, "gtdblib.taxon.rank.TaxonRank.GENUS"]], "order (gtdblib.taxon.rank.taxonrank attribute)": [[5, "gtdblib.taxon.rank.TaxonRank.ORDER"]], "phylum (gtdblib.taxon.rank.taxonrank attribute)": [[5, "gtdblib.taxon.rank.TaxonRank.PHYLUM"]], "species (gtdblib.taxon.rank.taxonrank attribute)": [[5, "gtdblib.taxon.rank.TaxonRank.SPECIES"]], "taxon (class in gtdblib.taxon.taxon)": [[5, "gtdblib.taxon.taxon.Taxon"]], "taxonrank (class in gtdblib.taxon.rank)": [[5, "gtdblib.taxon.rank.TaxonRank"]], "gtdblib.taxon": [[5, "module-gtdblib.taxon"]], "gtdblib.taxon.rank": [[5, "module-gtdblib.taxon.rank"]], "gtdblib.taxon.taxon": [[5, "module-gtdblib.taxon.taxon"]], "rank (gtdblib.taxon.taxon.taxon attribute)": [[5, "gtdblib.taxon.taxon.Taxon.rank"]], "value (gtdblib.taxon.taxon.taxon attribute)": [[5, "gtdblib.taxon.taxon.Taxon.value"]], "class_index (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.CLASS_INDEX"]], "domain_index (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.DOMAIN_INDEX"]], "family_index (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.FAMILY_INDEX"]], "genus_index (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.GENUS_INDEX"]], "order_index (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.ORDER_INDEX"]], "phylum_index (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.PHYLUM_INDEX"]], "rank_index (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.RANK_INDEX"]], "rank_labels (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.RANK_LABELS"]], "rank_prefixes (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.RANK_PREFIXES"]], "species_index (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.SPECIES_INDEX"]], "taxonomy (class in gtdblib.taxonomy.taxonomy)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy"]], "c (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.c"]], "d (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.d"]], "duplicate_names() (in module gtdblib.taxonomy.validation)": [[6, "gtdblib.taxonomy.validation.duplicate_names"]], "f (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.f"]], "fill_trailing_ranks() (in module gtdblib.taxonomy.taxonomy)": [[6, "gtdblib.taxonomy.taxonomy.fill_trailing_ranks"]], "g (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.g"]], "gtdblib.taxonomy": [[6, "module-gtdblib.taxonomy"]], "gtdblib.taxonomy.taxonomy": [[6, "module-gtdblib.taxonomy.taxonomy"]], "gtdblib.taxonomy.validation": [[6, "module-gtdblib.taxonomy.validation"]], "o (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.o"]], "p (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.p"]], "read_taxonomy() (in module gtdblib.taxonomy.taxonomy)": [[6, "gtdblib.taxonomy.taxonomy.read_taxonomy"]], "read_taxonomy_from_tree() (in module gtdblib.taxonomy.taxonomy)": [[6, "gtdblib.taxonomy.taxonomy.read_taxonomy_from_tree"]], "s (gtdblib.taxonomy.taxonomy.taxonomy attribute)": [[6, "gtdblib.taxonomy.taxonomy.Taxonomy.s"]], "taxonomic_consistency() (in module gtdblib.taxonomy.validation)": [[6, "gtdblib.taxonomy.validation.taxonomic_consistency"]], "validate_species_name() (in module gtdblib.taxonomy.validation)": [[6, "gtdblib.taxonomy.validation.validate_species_name"]], "validate_taxonomy() (in module gtdblib.taxonomy.validation)": [[6, "gtdblib.taxonomy.validation.validate_taxonomy"]], "bootstrap_merge_replicates() (in module gtdblib.tree.bootstrap_merge)": [[7, "gtdblib.tree.bootstrap_merge.bootstrap_merge_replicates"]], "collapse_polytomy() (in module gtdblib.tree.polytomy)": [[7, "gtdblib.tree.polytomy.collapse_polytomy"]], "collapse_polytomy_tree() (in module gtdblib.tree.polytomy)": [[7, "gtdblib.tree.polytomy.collapse_polytomy_tree"]], "convert_tree_accessions_to_canonical() (in module gtdblib.tree.convert_accession)": [[7, "gtdblib.tree.convert_accession.convert_tree_accessions_to_canonical"]], "get_tree_node_depth() (in module gtdblib.tree.get_node_depth)": [[7, "gtdblib.tree.get_node_depth.get_tree_node_depth"]], "get_tree_node_to_desc_taxa() (in module gtdblib.tree.get_node_to_desc_taxa)": [[7, "gtdblib.tree.get_node_to_desc_taxa.get_tree_node_to_desc_taxa"]], "get_tree_unique_mrca_for_internal_nodes() (in module gtdblib.tree.get_unique_mrca_for_internal_nodes)": [[7, "gtdblib.tree.get_unique_mrca_for_internal_nodes.get_tree_unique_mrca_for_internal_nodes"]], "gtdblib.tree": [[7, "module-gtdblib.tree"]], "gtdblib.tree.bootstrap_merge": [[7, "module-gtdblib.tree.bootstrap_merge"]], "gtdblib.tree.convert_accession": [[7, "module-gtdblib.tree.convert_accession"]], "gtdblib.tree.get_node_depth": [[7, "module-gtdblib.tree.get_node_depth"]], "gtdblib.tree.get_node_to_desc_taxa": [[7, "module-gtdblib.tree.get_node_to_desc_taxa"]], "gtdblib.tree.get_unique_mrca_for_internal_nodes": [[7, "module-gtdblib.tree.get_unique_mrca_for_internal_nodes"]], "gtdblib.tree.polytomy": [[7, "module-gtdblib.tree.polytomy"]], "gtdblib.util": [[8, "module-gtdblib.util"]], "gtdblib.util.color": [[8, "module-gtdblib.util.color"]], "gtdblib.util.types": [[8, "module-gtdblib.util.types"]], "is_float() (in module gtdblib.util.types)": [[8, "gtdblib.util.types.is_float"]], "rgb_to_hex() (in module gtdblib.util.color)": [[8, "gtdblib.util.color.rgb_to_hex"]], "bioerror": [[9, "gtdblib.util.bio.exceptions.BioError"]], "inputfileerror": [[9, "gtdblib.util.bio.seq_io.InputFileError"]], "canonical_gid() (in module gtdblib.util.bio.accession)": [[9, "gtdblib.util.bio.accession.canonical_gid"]], "create_label() (in module gtdblib.util.bio.newick)": [[9, "gtdblib.util.bio.newick.create_label"]], "extract_seqs() (in module gtdblib.util.bio.seq_io)": [[9, "gtdblib.util.bio.seq_io.extract_seqs"]], "gtdblib.util.bio": [[9, "module-gtdblib.util.bio"]], "gtdblib.util.bio.accession": [[9, "module-gtdblib.util.bio.accession"]], "gtdblib.util.bio.exceptions": [[9, "module-gtdblib.util.bio.exceptions"]], "gtdblib.util.bio.newick": [[9, "module-gtdblib.util.bio.newick"]], "gtdblib.util.bio.seq_io": [[9, "module-gtdblib.util.bio.seq_io"]], "is_nucleotide() (in module gtdblib.util.bio.seq_io)": [[9, "gtdblib.util.bio.seq_io.is_nucleotide"]], "is_protein() (in module gtdblib.util.bio.seq_io)": [[9, "gtdblib.util.bio.seq_io.is_protein"]], "is_same_accn_version() (in module gtdblib.util.bio.accession)": [[9, "gtdblib.util.bio.accession.is_same_accn_version"]], "parse_label() (in module gtdblib.util.bio.newick)": [[9, "gtdblib.util.bio.newick.parse_label"]], "read() (in module gtdblib.util.bio.seq_io)": [[9, "gtdblib.util.bio.seq_io.read"]], "read_fasta() (in module gtdblib.util.bio.seq_io)": [[9, "gtdblib.util.bio.seq_io.read_fasta"]], "read_fasta_seq() (in module gtdblib.util.bio.seq_io)": [[9, "gtdblib.util.bio.seq_io.read_fasta_seq"]], "read_fastq() (in module gtdblib.util.bio.seq_io)": [[9, "gtdblib.util.bio.seq_io.read_fastq"]], "read_fastq_seq() (in module gtdblib.util.bio.seq_io)": [[9, "gtdblib.util.bio.seq_io.read_fastq_seq"]], "read_seq() (in module gtdblib.util.bio.seq_io)": [[9, "gtdblib.util.bio.seq_io.read_seq"]], "seq_lengths() (in module gtdblib.util.bio.seq_io)": [[9, "gtdblib.util.bio.seq_io.seq_lengths"]], "write_fasta() (in module gtdblib.util.bio.seq_io)": [[9, "gtdblib.util.bio.seq_io.write_fasta"]], "check_dependencies() (in module gtdblib.util.shell.execute)": [[10, "gtdblib.util.shell.execute.check_dependencies"]], "check_dir_exists() (in module gtdblib.util.shell.gtdbshutil)": [[10, "gtdblib.util.shell.gtdbshutil.check_dir_exists"]], "check_file_exists() (in module gtdblib.util.shell.gtdbshutil)": [[10, "gtdblib.util.shell.gtdbshutil.check_file_exists"]], "check_on_path() (in module gtdblib.util.shell.execute)": [[10, "gtdblib.util.shell.execute.check_on_path"]], "get_num_lines() (in module gtdblib.util.shell.gtdbshutil)": [[10, "gtdblib.util.shell.gtdbshutil.get_num_lines"]], "gtdblib.util.shell": [[10, "module-gtdblib.util.shell"]], "gtdblib.util.shell.color": [[10, "module-gtdblib.util.shell.color"]], "gtdblib.util.shell.execute": [[10, "module-gtdblib.util.shell.execute"]], "gtdblib.util.shell.filemgmt": [[10, "module-gtdblib.util.shell.filemgmt"]], "gtdblib.util.shell.gtdbshutil": [[10, "module-gtdblib.util.shell.gtdbshutil"]], "hex_to_rgb() (in module gtdblib.util.shell.color)": [[10, "gtdblib.util.shell.color.hex_to_rgb"]], "is_executable() (in module gtdblib.util.shell.execute)": [[10, "gtdblib.util.shell.execute.is_executable"]], "make_sure_path_exists() (in module gtdblib.util.shell.gtdbshutil)": [[10, "gtdblib.util.shell.gtdbshutil.make_sure_path_exists"]], "matching_brackets() (in module gtdblib.util.shell.filemgmt)": [[10, "gtdblib.util.shell.filemgmt.matching_brackets"]], "rgb_to_hex() (in module gtdblib.util.shell.color)": [[10, "gtdblib.util.shell.color.rgb_to_hex"]], "run() (in module gtdblib.util.shell.execute)": [[10, "gtdblib.util.shell.execute.run"]], "select_delimiter() (in module gtdblib.util.shell.filemgmt)": [[10, "gtdblib.util.shell.filemgmt.select_delimiter"]], "sha256() (in module gtdblib.util.shell.filemgmt)": [[10, "gtdblib.util.shell.filemgmt.sha256"]], "sha256_rb() (in module gtdblib.util.shell.filemgmt)": [[10, "gtdblib.util.shell.filemgmt.sha256_rb"]], "symlink() (in module gtdblib.util.shell.gtdbshutil)": [[10, "gtdblib.util.shell.gtdbshutil.symlink"]], "which() (in module gtdblib.util.shell.execute)": [[10, "gtdblib.util.shell.execute.which"]]}})